---
title: "Mapping results for S. sinensis and S. sanguinis in ancient dental calculus"
author: "Irina Velsko"
date: "`r format(Sys.time(), '%b %d, %Y')`"
output:
    github_document:
        fig_width: 8
        fig_height: 6
        toc: true
---

```{r load_libraries, echo = F, message = F}
library(knitr)
library(data.table)
library(janitor)
library(dtplyr)
library(tidyverse)
library(patchwork)
opts_chunk$set(echo=F, warning=F, message=F, dpi=150)
```

Set the working directory of the notebook to the parent directory of the 
notebook itself.  

```{r setup}
knitr::opts_knit$set(root.dir = normalizePath(".."))
```

```{r}
sp_colors <- c("sanguinis" = "#ffffd9", "sanguinis_A" = "#edf8b1", "sanguinis_C" = "#c7e9b4", "sanguinis_D" = "#7fcdbb", "sanguinis_E" = "#41b6c4", "sanguinis_F" = "#1d91c0", "sanguinis_G" = "#225ea8", "sanguinis_H" = "#253494", "sanguinis_I" = "#081d58", "sinensis" = "#C70E7B")

```


```{r load_metadata}

genome_metadata <- fread("./00-documentation/sinensis_sanguinis_contigs.tsv") %>%
  as_tibble()

# also need to normalize the number of SNPs by number of bps in each genome
genome_lengths <- fread("./00-documentation/S_sinensis_sanguinis.fna.fai", sep = "\t", header = FALSE) %>%
  as_tibble() %>%
  rename(Contig = 1,
         Length = 2,
         TotalLen = 3,
         Offset = 4,
         Offset_1 = 5) %>%
  full_join(., genome_metadata %>%
              select(Contig, Assembly_accession), by = "Contig") %>%
  group_by(Assembly_accession) %>%
  summarize(GenomeLen = sum(Length)) %>%
  ungroup()

sp_order <- genome_lengths %>%
  left_join(., genome_metadata, by = "Assembly_accession") %>%
  select(Assembly_accession, GTDB_species) %>%
  distinct() %>%
  arrange(GTDB_species) %>%
  pull(Assembly_accession)


#  
lib_metadata <- fread("./00-documentation/strep_clades_metadata.tsv") %>%
  as_tibble() %>%
  filter(!str_detect(Study, "ObregonTito2015|Oh2016|Rampelli2015|Sankaranarayanan2015|Slon2017")) %>%
    filter(Sample_group != "non_human_primate",
           !str_detect(Sample_group,"urban_gut|blank|Bone")
           ) %>%
  mutate(Secondary_sample_accession = ifelse(Study == "Velsko2018", Sample_alias,Secondary_sample_accession),
         Secondary_sample_accession = ifelse(Study == "Velsko2023", Sample_alias,Secondary_sample_accession),
         Secondary_sample_accession = ifelse(Study == "Velsko2022", Sample_alias,Secondary_sample_accession),
         Secondary_sample_accession = ifelse(Study == "FellowsYates2021", Sample_alias,Secondary_sample_accession),
         Secondary_sample_accession = ifelse(Study == "Clemente2015", Run_accession,Secondary_sample_accession),
         Secondary_sample_accession = ifelse(Study == "Lassalle2017", Run_accession,Secondary_sample_accession),
         Secondary_sample_accession = ifelse(Study == "Brealey2020", Sample_alias,Secondary_sample_accession),
         Secondary_sample_accession = ifelse(Study == "Eerkens2018", Run_accession,Secondary_sample_accession),
         Secondary_sample_accession = ifelse(Study == "Eisenhoffer2020", Run_accession,Secondary_sample_accession),
         Secondary_sample_accession = ifelse(Study == "Fagernaes2022", Sample_alias,Secondary_sample_accession),
         Secondary_sample_accession = ifelse(Study == "Granehaell2021", Sample_alias,Secondary_sample_accession),
         Secondary_sample_accession = ifelse(Study == "Jacobson2020", Sample_alias,Secondary_sample_accession),
         Secondary_sample_accession = ifelse(Study == "Mann2018", Sample_alias,Secondary_sample_accession),
         Secondary_sample_accession = ifelse(Study == "Mann2018_w", Run_accession,Secondary_sample_accession),
         Secondary_sample_accession = ifelse(Study == "Ottoni2019", Sample_alias,Secondary_sample_accession),
         Secondary_sample_accession = ifelse(Study == "Ottoni2021", Run_accession,Secondary_sample_accession),
         Secondary_sample_accession = ifelse(Study == "Ozga2019", Sample_alias,Secondary_sample_accession),
         Secondary_sample_accession = ifelse(Study == "Ottoni2021", Run_accession,Secondary_sample_accession),
         Secondary_sample_accession = ifelse(Study == "Iberian", Sample_alias,Secondary_sample_accession),
         Secondary_sample_accession = ifelse(Study == "Pacific", Sample_alias,Secondary_sample_accession),
         Secondary_sample_accession = ifelse(Study == "Weyrich2017", Run_accession,Secondary_sample_accession),
         Secondary_sample_accession = ifelse(Study == "Warinner2014", Sample_alias,Secondary_sample_accession),
         Secondary_sample_accession = ifelse(Study == "Neukamm2020", Sample_alias,Secondary_sample_accession)) %>%
  mutate(Secondary_sample_accession = str_replace_all(Secondary_sample_accession, ".SG1.1",""))

# also read in cmc metadata for plots later
cmc_metadata <- fread("../Cameroon_plaque/00-documentation.backup/01-cameroon_hunter_gatherer_metadata.tsv") %>%
  as_tibble() %>%
  rename(Library_ID = 1) %>%
  filter(Env == "CMC") %>%
  select(Library_ID, Ethnic_Group, Market_integration, Sex, Tooth_site)

```

```{r}
# file with species names and clade designations
gtdb_clades <- fread("./00-documentation/dRep_gtdb_clades.tsv") %>%
  as_tibble()

gtdb_clades <- gtdb_clades %>%
  mutate(Clade = fct_relevel(Clade, "Sanguinis","Mitis","Salivarius","Anginosus","Bovis","Pyogenic","Mutans","Downei","Other","Unknown"))

```


## Ancient calculus
```{r load_data}
# read in the species table. It has the column names already
anc_calc_raw <- fread("./05-results/anc_calc.kraken2.gtdb.report_mpa.tsv.gz") %>%
  as_tibble()

anc_names <- fread("./05-results/anc_calc_names.tsv", header = F)  %>%
  as_tibble() %>%
  rename(Library_ID = 1) %>%
  mutate(Library_ID = str_replace_all(Library_ID, ".SG1.1", ""),
         Library_ID = str_replace_all(Library_ID, ".SG1", ""))

anc_calc_strep_sp_pre <- anc_calc_raw %>% 
  filter(str_detect(Lineage, "s__s__")) %>%
  separate(Lineage, into = c("domain","phylum","class","order","family","genus","Species"), sep = "\\|", extra = "merge") %>%
  select(-c("domain","phylum","class","order","family","genus")) %>%
  mutate(Species = str_replace_all(Species, "s__s__","")) %>%
  filter(str_detect(Species, "Streptococcus"))

# filter to have only Streptococcus
# these samples have very, very few reads that were assigned to the Sanguinis group (0-36), so we'll remove them
ac_strep_multiples <- anc_calc_strep_sp_pre %>%
  pivot_longer(!Species, names_to = "Library_ID", values_to = "Counts") %>%
  left_join(., gtdb_clades, by = "Species") %>%
  filter(str_detect(Clade, "Sanguinis")) %>%
  group_by(Library_ID) %>%
  slice_max(Counts) %>%
  ungroup() %>%
  select(-Counts, -Clade) %>%
  mutate(Species = str_replace_all(Species, "Streptococcus ", "")) %>%
  group_by(Library_ID) %>%
  count() %>% 
  ungroup() %>%
  filter(n > 1) %>%
  pull(Library_ID)

anc_calc_strep_sp <- anc_calc_raw %>% 
  filter(str_detect(Lineage, "s__s__")) %>%
  separate(Lineage, into = c("domain","phylum","class","order","family","genus","Species"), sep = "\\|", extra = "merge") %>%
  select(-c("domain","phylum","class","order","family","genus")) %>%
  mutate(Species = str_replace_all(Species, "s__s__","")) %>%
  filter(str_detect(Species, "Streptococcus")) %>%
  # remove Radcliffe soil and dentin
  select(-matches(ac_strep_multiples %>% str_c(collapse = "|")))

```

```{r}
# which samples have which strep species as the most abundant?
anc_calc_strep_sp %>%
  pivot_longer(!Species, names_to = "Library_ID", values_to = "Counts") %>%
  left_join(., gtdb_clades, by = "Species") %>%
  filter(str_detect(Clade, "Sanguinis")) %>%
  group_by(Library_ID) %>%
  slice_max(Counts) %>%
  ungroup() %>%
  arrange(Species) %>%
  group_by(Species) %>%
  count()

ac_most_abund_strep <- anc_calc_strep_sp %>%
  pivot_longer(!Species, names_to = "Library_ID", values_to = "Counts") %>%
  left_join(., gtdb_clades, by = "Species") %>%
  filter(str_detect(Clade, "Sanguinis")) %>%
  arrange(Library_ID, Species) %>%
  group_by(Library_ID) %>%
  slice_max(Counts) %>%
  ungroup() %>%
  select(-Counts, -Clade) %>%
  mutate(Species = str_replace_all(Species, "Streptococcus ", "")) 

```

```{r ac_load_cov, eval = F}

sss_list_ac <- list.files(path = "./04-analysis/sinensis_sanguinis_mapping/ancient_calculus/", pattern = ".cov", full.names = T)

sss_ac_cov <- map_dfr(sss_list_ac, function(fn) {
  fread(fn, col.names = c("Contig", "Depth", "Breadth","Genome_length","Proportion_covered"), sep = "\t") %>%
  mutate(Library_ID = basename(fn)) %>%
  as_tibble()
}) %>%
  arrange(Library_ID) %>%
  # remove the end of the file name from the Library_ID
  mutate(Library_ID = str_replace_all(Library_ID, "-sinensis_sanguinis_rep.mapped_rmdup.cov",""))

  fwrite(sss_ac_cov, "./05-results/sss_ac_cov.tsv", quote = F, sep = "\t")

```

```{r ac_load_cov}
sss_ac_cov <- fread("./05-results/sss_ac_cov.tsv.gz") %>%
  as_tibble()

# now add the genome accession to the table for filtering
sss_ac_cov <- sss_ac_cov %>%
  full_join(., genome_metadata, by = "Contig") %>%
  select(Assembly_accession, everything())


```

```{r ac_genome_coverages}

Genomes <- genome_metadata %>%
  select(Assembly_accession) %>%
  unique() %>%
  pull()

ac_genomes_coverage <- sss_ac_cov %>%
  filter(Assembly_accession != "all") %>%
  filter(Depth > 0) %>%
  group_by(Library_ID, Assembly_accession) %>%
  summarize(bases_cov_min1X = sum(Breadth)) %>%
  ungroup() %>%
  full_join(., sss_ac_cov %>%
              filter(Assembly_accession != "all") %>%
               select(Assembly_accession, Library_ID, Depth, Breadth) %>%
               filter(Depth >= 2) %>%
               group_by(Library_ID, Assembly_accession) %>%
               summarize(bases_cov_min2X = sum(Breadth)) %>%
               ungroup()) %>%
  full_join(., sss_ac_cov %>%
              filter(Assembly_accession != "all") %>%
               select(Assembly_accession, Library_ID, Depth, Breadth) %>%
               filter(Depth >= 5) %>%
               group_by(Library_ID, Assembly_accession) %>%
               summarize(bases_cov_min5X = sum(Breadth)) %>%
               ungroup()) %>%
  full_join(., genome_lengths, by = "Assembly_accession") %>%
  group_by(Library_ID, Assembly_accession) %>%
  summarize(breadth_1X = bases_cov_min1X / GenomeLen * 100,
            breadth_2X = bases_cov_min2X / GenomeLen * 100,
            breadth_5X = bases_cov_min5X / GenomeLen * 100) %>%
  ungroup() %>%
  mutate(Library_ID = str_replace_all(Library_ID, ".SG1", ""))

```

```{r ac_coverages}
# species with max breadth of coverage at 1X depth
ac_genomes_coverage %>%
  select(Library_ID, Assembly_accession, breadth_1X) %>%
  group_by(Library_ID) %>%
  slice_max(breadth_1X) %>%
  left_join(., genome_metadata %>%
              select(Assembly_accession, GTDB_species)) %>%
  distinct() %>%
  left_join(., ac_most_abund_strep, by = "Library_ID") %>%
  mutate(same_sp = GTDB_species == Species) %>%
  arrange(desc(same_sp), GTDB_species) %>%
  filter(Species == "sinensis")
  
# species with max breadth of coverage at 2X depth
ac_genomes_coverage %>%
  select(Library_ID, Assembly_accession, breadth_2X) %>%
  group_by(Library_ID) %>%
  slice_max(breadth_2X) %>%
  left_join(., genome_metadata %>%
              select(Assembly_accession, GTDB_species)) %>%
  distinct() %>%
  # remove these b/c they have no coverage at 2X depth
  filter(!str_detect(Library_ID, "DLV002.A0101|SPM001.A0102")) %>%
  left_join(., ac_most_abund_strep, by = "Library_ID") %>%
  mutate(same_sp = GTDB_species == Species) %>%
  arrange(desc(same_sp), GTDB_species) %>%
  filter(Species == "sinensis")

# species with max breadth of coverage at 5X depth
ac_genomes_coverage %>%
  select(Library_ID, Assembly_accession, breadth_5X) %>%
  group_by(Library_ID) %>%
  slice_max(breadth_5X) %>%
  left_join(., genome_metadata %>%
              select(Assembly_accession, GTDB_species)) %>%
  distinct() %>%
  # remove these b/c they have no coverage at 2X depth
  filter(!str_detect(Library_ID, "Abusir1504c|Abusir1511c|Abusir1574c|BAN001.A0101|BAN001.A0102|CS27|DLV001.A0101|DLV002.A0101|GUE001.A0102|HCLVMBCX2-3505-20-00-01_S20|HCLVMBCX2-3505-22-00-01_S22|HCLVMBCX2-3505-23-00-01_S23|HCLVMBCX2-3505-24-00-01_S24|HCLVMBCX2-3505-25-00-01_S25|HCLVMBCX2-3505-29-00-01_S29|HCLVMBCX2-3505-31-00-01_S31|HCLVMBCX2-3505-33-00-01_S33|SPM001.A0101|SPM001.A0102|SPM002.A0103|SPM003.A0101|SRR11176631|TAF016.B0101|TAF017.A0101|TAF017.E0101|ZAF001.A0103|ZAF001.B0101")) %>%
  left_join(., ac_most_abund_strep, by = "Library_ID") %>%
  mutate(same_sp = GTDB_species == Species) %>%
  arrange(desc(same_sp), GTDB_species) %>%
  filter(Species == "sinensis")
 

```

```{r}
ac_together_all <-  ac_genomes_coverage %>%
  pivot_longer(!Library_ID:Assembly_accession, names_to = "Depth", values_to = "Breadth") %>%
  filter(Depth != "breadth_2X") %>%
  left_join(., genome_metadata %>%
              select(Assembly_accession, GTDB_species)) %>%
  distinct() %>%
  full_join(., ac_most_abund_strep %>%
               separate(Species, into = "species", sep = "_", extra = "drop"), by = "Library_ID") %>%
  filter(str_detect(species, "sinensis|sanguinis")) %>%
  mutate(Assembly_accession = fct_relevel(Assembly_accession, sp_order)) %>%
  ggplot(., aes(x = Assembly_accession, y = Breadth, fill = GTDB_species)) + # Species
         geom_boxplot(outlier.colour = NA) +
         geom_jitter(aes(shape = species), size = 1, alpha = 0.5, width = 0.2) +
         scale_fill_manual(values = sp_colors) +
         scale_shape_manual(values = c(4, 1)) +
         # scale_shape_manual(values = c(8, 3, 6, 1)) +
         theme_minimal(base_size = 10) +
         theme(axis.text.x = element_text(angle = 90, hjust = 1, vjust = 0.5)) +
         ylim(0,100) +
         ylab("Breadth of genome covered (%)") +
         xlab("Assembly accession") +
  facet_wrap(~Depth)
ac_together_all

ggsave("./06-publication/supplemental_figures/Sup_fig_S16/ss_all_map_ac.pdf", plot = ac_together_all, device = "pdf",
       scale = 1, width = 8, height = 5, units = c("in"), dpi = 300)

```

```{r, eval = F}

ac_ss_together_all <- ac_together_all_1x / ac_together_all_5x +
  plot_layout(guides = "collect")

ac_ss_together_all

# ggsave("./06-publication/supplemental_figures/Sup_fig_S16/ss_all_map_ac.pdf", plot = ac_ss_together_all, device = "pdf",
#        scale = 1, width = 14, height = 9, units = c("in"), dpi = 300)

```


```{r cov_all_together}

ac_genomes_coverage %>%
  pivot_longer(!Library_ID:Assembly_accession, names_to = "Depth", values_to = "Breadth") %>%
  mutate(Library_ID = str_replace_all(Library_ID, ".SG1","")) %>%
  filter(Depth == "breadth_1X") %>%
  left_join(., genome_metadata %>%
              select(Assembly_accession, GTDB_species) %>%
              rename(Assembly_species = GTDB_species)) %>%
  distinct() %>%
  full_join(., ac_most_abund_strep %>%
               separate(Species, into = "species", sep = "_", extra = "drop"), by = "Library_ID") %>%
  filter(str_detect(species, "sinensis|sanguinis")) %>%
  drop_na(Breadth) %>%
  # get rid of the species cluster info so have only sanguinis/sinensis for plotting
  separate(Assembly_species, into = "Assembly_species", sep = "_", extra = "drop") %>%
  group_by(Library_ID, Assembly_species) %>%
  slice_max(Breadth, n = 1) %>%
  ungroup() %>%
  ggplot(.) +
         geom_point(aes(x = Assembly_species, y = Breadth, fill = species), shape = 21, color = "black") + # , fill = "#766df8"
         geom_line(aes(x = Assembly_species, y = Breadth, group = Library_ID), linetype="dotted") +
         theme_minimal(base_size = 12) +
         scale_fill_manual(values = sp_colors) +
         theme(axis.text.x = element_text(angle = 0, hjust = 0.5, size = 12),
               axis.text.y = element_text(size = 12)) +
         xlab("Genome") +
         ylab("% of genome covered at 1X") +
         facet_wrap(~species)


ac_genomes_coverage %>%
  pivot_longer(!Library_ID:Assembly_accession, names_to = "Depth", values_to = "Breadth") %>%
  mutate(Library_ID = str_replace_all(Library_ID, ".SG1","")) %>%
  filter(Depth == "breadth_1X") %>%
  left_join(., genome_metadata %>%
              select(Assembly_accession, GTDB_species) %>%
              rename(Assembly_species = GTDB_species)) %>%
  distinct() %>%
  full_join(., ac_most_abund_strep %>%
               separate(Species, into = "species", sep = "_", extra = "drop"), by = "Library_ID") %>%
  filter(str_detect(species, "sinensis|sanguinis")) %>%
  drop_na(Breadth) %>%
  # get rid of the species cluster info so have only sanguinis/sinensis for plotting
  separate(Assembly_species, into = "Assembly_species", sep = "_", extra = "drop") %>%
  group_by(Library_ID, Assembly_species) %>%
  slice_max(Breadth, n = 1) %>%
  ungroup() %>%
  select(-Assembly_accession) %>%
  pivot_wider(names_from = "Assembly_species", values_from = "Breadth") %>%
  mutate(Difference = sanguinis - sinensis,
         Slope = (sinensis - sanguinis) / (1-0)) %>%
  # filter(Slope >= 1 | Slope <= -1) %>%
  ggplot(., aes(x = species, y = Slope, fill = species)) + # Species
         geom_boxplot(outlier.colour = NA) +
         geom_jitter(aes(shape = species), alpha = 0.5, width = 0.2) +
         scale_fill_manual(values = sp_colors) +
         scale_shape_manual(values = c(4, 1)) +
         # scale_shape_manual(values = c(8, 3, 6, 1)) +
         theme_minimal(base_size = 12) +
         theme(axis.text.x = element_text(angle = 0, hjust = 0.5)) +
         geom_hline(yintercept = 0, linetype = "dashed") +
         ylab("Slope") +
         xlab("Species") 


```

```{r cov_all_independent}

ac_together_cov_plot <- ac_genomes_coverage %>%
  pivot_longer(!Library_ID:Assembly_accession, names_to = "Depth", values_to = "Breadth") %>%
  mutate(Library_ID = str_replace_all(Library_ID, ".SG1","")) %>%
  filter(Depth == "breadth_1X") %>%
  left_join(., genome_metadata %>%
              select(Assembly_accession, GTDB_species) %>%
              rename(Assembly_species = GTDB_species)) %>%
  distinct() %>%
  filter(str_detect(Assembly_accession, "013343115|000767835")) %>%
  full_join(., ac_most_abund_strep %>%
               separate(Species, into = "species", sep = "_", extra = "drop"), by = "Library_ID") %>%
  filter(str_detect(species, "sinensis|sanguinis")) %>%
  drop_na(Breadth) %>%
  # get rid of the species cluster info so have only sanguinis/sinensis for plotting
  separate(Assembly_species, into = "Assembly_species", sep = "_", extra = "drop") %>%
  ggplot(.) +
         geom_point(aes(x = Assembly_species, y = Breadth, fill = species), shape = 21, color = "black") + # , fill = "#766df8"
         geom_line(aes(x = Assembly_species, y = Breadth, group = Library_ID), linetype="dotted") +
         theme_minimal(base_size = 12) +
         scale_fill_manual(values = sp_colors) +
         theme(axis.text.x = element_text(angle = 0, hjust = 0.5, size = 12),
               axis.text.y = element_text(size = 12)) +
         xlab("Genome") +
         ylab("% of genome covered at 1X") +
         facet_wrap(~species)
ac_together_cov_plot

ac_together_cov_box_plot <- ac_genomes_coverage %>%
  pivot_longer(!Library_ID:Assembly_accession, names_to = "Depth", values_to = "Breadth") %>%
  mutate(Library_ID = str_replace_all(Library_ID, ".SG1","")) %>%
  filter(Depth == "breadth_1X") %>%
  left_join(., genome_metadata %>%
              select(Assembly_accession, GTDB_species) %>%
              rename(Assembly_species = GTDB_species)) %>%
  distinct() %>%
  filter(str_detect(Assembly_accession, "013343115|000767835")) %>%
  full_join(., ac_most_abund_strep %>%
               separate(Species, into = "species", sep = "_", extra = "drop"), by = "Library_ID") %>%
  filter(str_detect(species, "sinensis|sanguinis")) %>%
  drop_na(Breadth) %>%
  # get rid of the species cluster info so have only sanguinis/sinensis for plotting
  separate(Assembly_species, into = "Assembly_species", sep = "_", extra = "drop") %>%
  select(-Assembly_accession) %>%
  pivot_wider(names_from = "Assembly_species", values_from = "Breadth") %>%
  mutate(Difference = sanguinis - sinensis,
         Slope = (sinensis - sanguinis) / (1-0)) %>%
  # filter(Slope >= 1 | Slope <= -1) %>%
  ggplot(., aes(x = species, y = Slope, fill = species)) + # Species
         geom_boxplot(outlier.colour = NA) +
         geom_jitter(aes(shape = species), alpha = 0.5, width = 0.2) +
         scale_fill_manual(values = sp_colors) +
         scale_shape_manual(values = c(4, 1)) +
         # scale_shape_manual(values = c(8, 3, 6, 1)) +
         theme_minimal(base_size = 12) +
         theme(axis.text.x = element_text(angle = 0, hjust = 0.5, size = 12),
               axis.text.y = element_text(size = 12)) +
         geom_hline(yintercept = 0, linetype = "dashed") +
         ylab("Slope") +
         xlab("Species") 
ac_together_cov_box_plot

```

```{r}

ac_ss_together_map <- ac_together_cov_plot + ac_together_cov_box_plot +
  plot_layout(widths = c(1.5, 1)) +
  plot_layout(guides = "collect")

ac_ss_together_map

ggsave("./06-publication/supplemental_figures/Sup_fig_S16/ss_together_map_ac.pdf", plot = ac_ss_together_map, device = "pdf",
       scale = 1, width = 8, height = 5, units = c("in"), dpi = 300)

```

### Variants in vcf files
```{r, eval = F}

vcf_list_ac <- list.files(path = "./04-analysis/sinensis_sanguinis_mapping/ancient_calculus/", pattern = "sinensis_sanguinis.vcf$", full.names = T)

vcf_ac_snp <- map_dfr(vcf_list_ac, function(fn) {
  fread(fn, col.names = c("Contig", "POS", "ID","REF","ALT", "QUAL", "FILTER","INFO", "FORMAT","BAM"), sep = "\t", skip = 430, header = F) %>%
  mutate(Library_ID = basename(fn)) %>%
  as_tibble() %>%
  filter(FILTER == "PASS")
}) %>%
# remove the end of the file name from the Library_ID
  mutate(Library_ID = str_replace_all(Library_ID, ".sinensis_sanguinis.vcf","")) %>%
  mutate(Library_ID = str_replace_all(Library_ID, ".SG1","")) 

fwrite(vcf_ac_snp, "./05-results/ss_ac_vcf_snp.tsv", quote = F, sep = "\t")

```

```{r}
vcf_ac_snp <- fread("../strep_clades/05-results/ss_ac_vcf_snp.tsv.gz") %>%
  as_tibble()

vcf_ac_sum_snp  <- vcf_ac_snp %>%
  full_join(., genome_metadata, by = "Contig") %>%
  # remove any lines that are contigs with no mapping information
  filter(!is.na(POS)) %>%
  group_by(Library_ID, Assembly_accession) %>%
  count() %>%
  ungroup() %>%
  full_join(., genome_metadata %>%
              select(Assembly_accession, Species, GTDB_species), by = "Assembly_accession") %>%
  filter(str_detect(Assembly_accession, "GCF_013343115.1|GCF_000194945.1|GCF_000767835.1|GCF_001578725.1|GCF_019090945.1|GCF_902460385.1")) %>%
  distinct()

vcf_ac_2_snp <- vcf_ac_snp %>%
# remove the end of the file name from the Library_ID
  mutate(Library_ID = str_replace_all(Library_ID, ".sinensis_sanguinis.vcf","")) %>%
  mutate(Library_ID = str_replace_all(Library_ID, ".SG1","")) %>%
  full_join(., genome_metadata, by = "Contig") %>%
  # remove any lines that are contigs with no mapping information
  filter(!is.na(POS)) %>%
  distinct() %>%
  group_by(Library_ID, Assembly_accession) %>%
  ungroup() %>%
  separate(INFO, into = "Depth", extra = "drop", sep = ";") %>%
  mutate(Depth = str_replace_all(Depth, "DP=",""),
         Depth = as.numeric(Depth)) %>% 
  filter(Depth >= 2) %>%
  group_by(Library_ID, Assembly_accession) %>%
  count() %>%
  ungroup() %>%
  full_join(., genome_metadata %>%
              select(Assembly_accession, Species, GTDB_species), by = "Assembly_accession") %>%
  filter(str_detect(Assembly_accession, "GCF_013343115.1|GCF_000194945.1|GCF_000767835.1|GCF_001578725.1|GCF_019090945.1|GCF_902460385.1")) %>%
  distinct()
  

```


```{r}
vcf_ac_2_snp %>%
  mutate(Assembly_accession = fct_relevel(Assembly_accession, sp_order)) %>%
  left_join(., ac_most_abund_strep %>%
               separate(Species, into = "species", sep = "_", extra = "drop"), by = "Library_ID") %>%
  filter(str_detect(species, "sinensis|sanguinis")) %>%
  ggplot(., aes(x = Assembly_accession, y = n, fill = GTDB_species)) + # Species
         geom_boxplot(outlier.colour = NA) +
         geom_jitter(aes(shape = species), alpha = 0.5, width = 0.2) +
         scale_fill_manual(values = sp_colors) +
         scale_shape_manual(values = c(4, 1)) +
         # scale_shape_manual(values = c(8, 3, 6, 1)) +
         theme_minimal(base_size = 10) +
         theme(axis.text.x = element_text(angle = 90, hjust = 1)) +
         xlab("Assembly accession") +
         ylab("Total SNPs/Genome, depth >2") +


vcf_ac_2_snp %>%
  left_join(., ac_most_abund_strep %>%
               separate(Species, into = "species", sep = "_", extra = "drop"), by = "Library_ID") %>%
  filter(str_detect(species, "sinensis|sanguinis")) %>%
  left_join(., genome_lengths) %>%
  mutate(proportion_snps = n / GenomeLen,
         Assembly_accession = fct_relevel(Assembly_accession, sp_order)) %>%
  arrange(Assembly_accession) %>%
  ggplot(., aes(x = Assembly_accession, y = proportion_snps, fill = GTDB_species)) +
         geom_boxplot(outlier.colour = NA) +
         geom_jitter(aes(shape = species), alpha = 0.5, width = 0.2) +
         scale_fill_manual(values = sp_colors) +
         scale_shape_manual(values = c(4, 1)) +
         # scale_shape_manual(values = c(8, 3, 6, 1)) +
         theme_minimal(base_size = 10) +
         theme(axis.text.x = element_text(angle = 90, hjust = 1)) +
         xlab("Assembly accession") +
         ylab("% SNPs/Genome")


vcf_ac_sum_snp %>%
  left_join(., ac_most_abund_strep %>%
               separate(Species, into = "species", sep = "_", extra = "drop"), by = "Library_ID") %>%
  filter(str_detect(species, "sinensis|sanguinis")) %>%
  mutate(Assembly_accession = fct_relevel(Assembly_accession, sp_order)) %>%
  ggplot(., aes(x = Assembly_accession, y = n, fill = GTDB_species)) + # Species
         geom_boxplot(outlier.colour = NA) +
         geom_jitter(aes(shape = species), alpha = 0.5, width = 0.2) +
         scale_fill_manual(values = sp_colors) +
         scale_shape_manual(values = c(4, 1)) +
         # scale_shape_manual(values = c(8, 3, 6, 1)) +
         theme_minimal(base_size = 10) +
         theme(axis.text.x = element_text(angle = 90, hjust = 1)) +
         xlab("Assembly accession") +
         ylab("Total SNPs/Genome")


vcf_ac_sum_snp %>%
  left_join(., ac_most_abund_strep %>%
               separate(Species, into = "species", sep = "_", extra = "drop"), by = "Library_ID") %>%
  filter(str_detect(species, "sinensis|sanguinis")) %>%
  left_join(., genome_lengths) %>%
  mutate(proportion_snps = n / GenomeLen,
         Assembly_accession = fct_relevel(Assembly_accession, sp_order)) %>%
  arrange(Assembly_accession) %>%
  ggplot(., aes(x = Assembly_accession, y = proportion_snps, fill = GTDB_species)) +
         geom_boxplot(outlier.colour = NA) +
         geom_jitter(aes(shape = species), alpha = 0.5, width = 0.2) +
         scale_fill_manual(values = sp_colors) +
         scale_shape_manual(values = c(4, 1)) +
         # scale_shape_manual(values = c(8, 3, 6, 1)) +
         theme_minimal(base_size = 10) +
         theme(axis.text.x = element_text(angle = 90, hjust = 1)) +
         xlab("Assembly accession") +
         ylab("% SNPs/Genome")


# ./04-analysis/sinensis_sanguinis_mapping/non_ind_plaque/CMC002.A0201.SG1.sinensis_sanguinis.vcf

# maybe make this a alluvial plot with the left side sinensis SNP counts and the right side Sanguinis SNP counts?
# Do samples with higher sanguinis SNP counts have lower sinensis SNP counts?

```

```{r}

# may need to facet-wrap this one by Study
vcf_ac_sum_snp %>%
  left_join(., genome_lengths) %>%
  filter(str_detect(Assembly_accession, "013343115|000767835")) %>%
  left_join(., lib_metadata %>%
              select(Secondary_sample_accession, Study) %>%
              rename(Library_ID = Secondary_sample_accession), by = "Library_ID") %>%
  mutate(proportion_snps = n / GenomeLen,
         Assembly_accession = fct_relevel(Assembly_accession, "GCF_013343115.1","GCF_000767835.1")) %>%
  arrange(Assembly_accession) %>%
  ggplot(.) +
         geom_point(aes(x = GTDB_species, y = proportion_snps), shape = 21, color = "black", fill = "#766df8") +
         geom_line(aes(x = GTDB_species, y = proportion_snps, group = Library_ID), linetype="dotted") +
         theme_minimal(base_size = 8) +
         theme(axis.text.x = element_text(angle = 90, hjust = 0.5, size = 8),
               axis.text.y = element_text(size = 8)) +
         xlab("Genome") +
         ylab("% SNPs/Genome") +
         facet_wrap(~Study)

# want to compare coverage depth/breadth here instead of % SNPs
# use a different input table
# Is there some metric that could be used to measure both depth and breadth together?
# Something like alpha-diversity?
ac_genomes_coverage %>%
  pivot_longer(!Library_ID:Assembly_accession, names_to = "Depth", values_to = "Breadth") %>%
  mutate(Library_ID = str_replace_all(Library_ID, ".SG1","")) %>%
  filter(Depth == "breadth_1X") %>%
  left_join(., genome_metadata %>%
              select(Assembly_accession, GTDB_species)) %>%
  distinct() %>%
  filter(str_detect(Assembly_accession, "013343115|000767835")) %>%
  left_join(., lib_metadata %>%
              select(Secondary_sample_accession, Study) %>%
              rename(Library_ID = Secondary_sample_accession), by = "Library_ID") %>%
  mutate(Assembly_accession = fct_relevel(Assembly_accession, "GCF_003943655.1","GCF_000767835.1")) %>%
  ggplot(.) +
         geom_point(aes(x = Assembly_accession, y = Breadth), shape = 21, color = "black", fill = "#766df8") +
         geom_line(aes(x = Assembly_accession, y = Breadth, group = Library_ID), linetype="dotted") +
         theme_minimal(base_size = 8) +
         theme(axis.text.x = element_text(angle = 90, hjust = 0.5, size = 8),
               axis.text.y = element_text(size = 8)) +
         xlab("Genome") +
         ylab("% of genome covered at 1X") +
         facet_wrap(~Study)

```

### Independent mappings

```{r ac_load_independent_cov, eval = F}
# Sinensis
sinensis_list_ac <- list.files(path = "./04-analysis/sinensis_sanguinis_mapping/ancient_calculus/sinensis", pattern = ".cov", full.names = T)

sinensis_ac_cov <- map_dfr(sinensis_list_ac, function(fn) {
  fread(fn, col.names = c("Contig", "Depth", "Breadth","Genome_length","Proportion_covered"), sep = "\t") %>%
  mutate(Library_ID = basename(fn)) %>%
  as_tibble()
}) %>%
  arrange(Library_ID) %>%
# remove the end of the file name from the Library_ID
  mutate(Library_ID = str_replace_all(Library_ID, "-sinensis.mapped_rmdup.cov",""))
  
# fwrite(sinensis_ac_cov, "05-results/sinensis_ac_indep_cov.tsv", quote = F, sep = "\t")

# Sanguinis
sanguinis_list_cmc <- list.files(path = "./04-analysis/sinensis_sanguinis_mapping/ancient_calculus/sanguinis", pattern = ".cov", full.names = T)

sanguinis_ac_cov <- map_dfr(sanguinis_list_cmc, function(fn) {
  fread(fn, col.names = c("Contig", "Depth", "Breadth","Genome_length","Proportion_covered"), sep = "\t") %>%
  mutate(Library_ID = basename(fn)) %>%
  as_tibble()
}) %>%
  arrange(Library_ID) %>%
# remove the end of the file name from the Library_ID
  mutate(Library_ID = str_replace_all(Library_ID, "-sanguinis.mapped_rmdup.cov",""))  

# fwrite(sanguinis_ac_cov, "05-results/sanguinis_ac_indep_cov.tsv", quote = F, sep = "\t")

```

```{r ac_load_independent_cov}
sinensis_ac_cov <- fread("../strep_clades/05-results/sinensis_ac_indep_cov.tsv.gz") %>%
  as_tibble()
sanguinis_ac_cov <- fread("../strep_clades/05-results/sanguinis_ac_indep_cov.tsv.gz") %>%
  as_tibble()

# now add the genome accession to the table for filtering
sinensis_ac_cov <- sinensis_ac_cov %>%
  left_join(., genome_metadata, by = "Contig") %>%
  select(Assembly_accession, everything()) %>%
  filter(Contig != "genome")


# now add the genome accession to the table for filtering
sanguinis_ac_cov <- sanguinis_ac_cov %>%
  left_join(., genome_metadata, by = "Contig") %>%
  select(Assembly_accession, everything()) %>%
  filter(Contig != "genome")

```


```{r ac_independent_coverages}

ac_independent_coverages <-  sinensis_ac_cov %>%
  filter(Assembly_accession != "all") %>%
  filter(Depth > 0) %>%
  group_by(Library_ID, Assembly_accession) %>%
  summarize(bases_cov_min1X = sum(Breadth)) %>%
  ungroup() %>%
  full_join(., sinensis_ac_cov %>%
              filter(Assembly_accession != "all") %>%
               select(Assembly_accession, Library_ID, Depth, Breadth) %>%
               filter(Depth >= 2) %>%
               group_by(Library_ID, Assembly_accession) %>%
               summarize(bases_cov_min2X = sum(Breadth)) %>%
               ungroup()) %>%
  full_join(., sinensis_ac_cov %>%
              filter(Assembly_accession != "all") %>%
               select(Assembly_accession, Library_ID, Depth, Breadth) %>%
               filter(Depth >= 5) %>%
               group_by(Library_ID, Assembly_accession) %>%
               summarize(bases_cov_min5X = sum(Breadth)) %>%
               ungroup()) %>%
  full_join(., genome_lengths, by = "Assembly_accession") %>%
  group_by(Library_ID, Assembly_accession) %>%
  summarize(breadth_1X = bases_cov_min1X / GenomeLen * 100,
            breadth_2X = bases_cov_min2X / GenomeLen * 100,
            breadth_5X = bases_cov_min5X / GenomeLen * 100) %>%
  ungroup() %>%
  bind_rows(., sanguinis_ac_cov %>%
  filter(Assembly_accession != "all") %>%
  filter(Depth > 0) %>%
  group_by(Library_ID, Assembly_accession) %>%
  summarize(bases_cov_min1X = sum(Breadth)) %>%
  ungroup() %>%
  full_join(., sanguinis_ac_cov %>%
              filter(Assembly_accession != "all") %>%
               select(Assembly_accession, Library_ID, Depth, Breadth) %>%
               filter(Depth >= 2) %>%
               group_by(Library_ID, Assembly_accession) %>%
               summarize(bases_cov_min2X = sum(Breadth)) %>%
               ungroup()) %>%
  full_join(., sanguinis_ac_cov %>%
              filter(Assembly_accession != "all") %>%
               select(Assembly_accession, Library_ID, Depth, Breadth) %>%
               filter(Depth >= 5) %>%
               group_by(Library_ID, Assembly_accession) %>%
               summarize(bases_cov_min5X = sum(Breadth)) %>%
               ungroup()) %>%
  full_join(., genome_lengths, by = "Assembly_accession") %>%
  group_by(Library_ID, Assembly_accession) %>%
  summarize(breadth_1X = bases_cov_min1X / GenomeLen * 100,
            breadth_2X = bases_cov_min2X / GenomeLen * 100,
            breadth_5X = bases_cov_min5X / GenomeLen * 100) %>%
  ungroup()) %>%
  pivot_longer(!Library_ID:Assembly_accession, names_to = "Depth", values_to = "Breadth")


```

```{r}
ac_independent_coverages %>%
  filter(Depth == "breadth_1X") %>%
  left_join(., ac_most_abund_strep %>%
               separate(Species, into = "species", sep = "_", extra = "drop"), by = "Library_ID") %>%
  filter(str_detect(species, "sinensis|sanguinis")) %>%
  left_join(., genome_metadata %>%
              select(Assembly_accession, GTDB_species)) %>%
  distinct() %>%
  mutate(Assembly_accession = fct_relevel(Assembly_accession, sp_order)) %>%
  ggplot(., aes(x = Assembly_accession, y = Breadth, fill = GTDB_species)) + # Species
         geom_boxplot(outlier.colour = NA) +
         geom_jitter(aes(shape = species), alpha = 0.5, width = 0.2) +
         scale_fill_manual(values = sp_colors) +
         scale_shape_manual(values = c(4, 1)) +
         # scale_shape_manual(values = c(8, 3, 6, 1)) +
         theme_minimal(base_size = 10) +
         theme(axis.text.x = element_text(angle = 90, hjust = 1)) +
         ylim(0,100) +
         xlab("Assembly accession") +
         ylab("Breadth of genome covered at least 1X (%)") +
         ggtitle("Independent genome mapping")

ac_independent_coverages %>%
  filter(Depth == "breadth_5X") %>%
  left_join(., ac_most_abund_strep %>%
               separate(Species, into = "species", sep = "_", extra = "drop"), by = "Library_ID") %>%
  filter(str_detect(species, "sinensis|sanguinis")) %>%
  left_join(., genome_metadata %>%
              select(Assembly_accession, GTDB_species)) %>%
  distinct() %>%
  # remove samples with no coverage at 5X
  drop_na(Breadth) %>%
  mutate(Assembly_accession = fct_relevel(Assembly_accession, sp_order)) %>%
  ggplot(., aes(x = Assembly_accession, y = Breadth, fill = GTDB_species)) + # Species
         geom_boxplot(outlier.colour = NA) +
         geom_jitter(aes(shape = species), alpha = 0.5, width = 0.2) +
         scale_fill_manual(values = sp_colors) +
         scale_shape_manual(values = c(4, 1)) +
         # scale_shape_manual(values = c(8, 3, 6, 1)) +
         theme_minimal(base_size = 10) +
         theme(axis.text.x = element_text(angle = 90, hjust = 1)) +
         ylim(0,100) +
         xlab("Assembly accession") +
         ylab("Breadth of genome covered at least 5X (%)") +
         ggtitle("Independent genome mapping")

```

```{r cov_independent}

ac_indep_cov_plot <-  ac_independent_coverages %>%
  mutate(Library_ID = str_replace_all(Library_ID, ".SG1","")) %>%
  filter(Depth == "breadth_1X") %>%
  left_join(., genome_metadata %>%
              select(Assembly_accession, GTDB_species) %>%
              rename(Assembly_species = GTDB_species)) %>%
  distinct() %>%
  full_join(., ac_most_abund_strep %>%
               separate(Species, into = "species", sep = "_", extra = "drop"), by = "Library_ID") %>%
  filter(str_detect(species, "sinensis|sanguinis")) %>%
  drop_na(Breadth) %>%
  # get rid of the species cluster info so have only sanguinis/sinensis for plotting
  separate(Assembly_species, into = "Assembly_species", sep = "_", extra = "drop") %>%
  ggplot(.) +
         geom_point(aes(x = Assembly_species, y = Breadth, fill = species), shape = 21, color = "black") + # , fill = "#766df8"
         geom_line(aes(x = Assembly_species, y = Breadth, group = Library_ID), linetype="dotted") +
         theme_minimal(base_size = 12) +
         scale_fill_manual(values = sp_colors) +
         theme(axis.text.x = element_text(angle = 0, hjust = 0.5, size = 12),
               axis.text.y = element_text(size = 12)) +
         xlab("Genome") +
         ylab("% of genome covered at 1X") +
         facet_wrap(~species)
ac_indep_cov_plot

ac_indep_cov_box_plot <- ac_independent_coverages %>%
  mutate(Library_ID = str_replace_all(Library_ID, ".SG1","")) %>%
  filter(Depth == "breadth_1X") %>%
  left_join(., genome_metadata %>%
              select(Assembly_accession, GTDB_species) %>%
              rename(Assembly_species = GTDB_species)) %>%
  distinct() %>%
  full_join(., ac_most_abund_strep %>%
               separate(Species, into = "species", sep = "_", extra = "drop"), by = "Library_ID") %>%
  filter(str_detect(species, "sinensis|sanguinis")) %>%
  drop_na(Breadth) %>%
  # get rid of the species cluster info so have only sanguinis/sinensis for plotting
  separate(Assembly_species, into = "Assembly_species", sep = "_", extra = "drop") %>%
  select(-Assembly_accession) %>%
  pivot_wider(names_from = "Assembly_species", values_from = "Breadth") %>%
  mutate(Difference = sinensis - sanguinis,
         Slope = (sinensis - sanguinis) / (1-0)) %>%
  # filter(Slope >= 1 | Slope <= -1) %>%
  ggplot(., aes(x = species, y = Slope, fill = species)) + # Species
         geom_boxplot(outlier.colour = NA) +
         geom_jitter(aes(shape = species), alpha = 0.5, width = 0.2) +
         scale_fill_manual(values = sp_colors) +
         scale_shape_manual(values = c(4, 1)) +
         # scale_shape_manual(values = c(8, 3, 6, 1)) +
         theme_minimal(base_size = 12) +
         theme(axis.text.x = element_text(angle = 0, hjust = 0.5, size = 12),
               axis.text.y = element_text(size = 12)) +
         geom_hline(yintercept = 0, linetype = "dashed") +
         ylab("Slope") +
         xlab("Species") 
ac_indep_cov_box_plot

ac_independent_coverages %>%
  mutate(Library_ID = str_replace_all(Library_ID, ".SG1","")) %>%
  filter(Depth == "breadth_1X") %>%
  left_join(., genome_metadata %>%
              select(Assembly_accession, GTDB_species) %>%
              rename(Assembly_species = GTDB_species)) %>%
  distinct() %>%
  full_join(., ac_most_abund_strep %>%
               separate(Species, into = "species", sep = "_", extra = "drop"), by = "Library_ID") %>%
  filter(str_detect(species, "sinensis|sanguinis")) %>%
  drop_na(Breadth) %>%
  # get rid of the species cluster info so have only sanguinis/sinensis for plotting
  separate(Assembly_species, into = "Assembly_species", sep = "_", extra = "drop") %>%
  select(-Assembly_accession) %>%
  pivot_wider(names_from = "Assembly_species", values_from = "Breadth") %>%
  mutate(Difference = sinensis - sanguinis,
         Slope = (sinensis - sanguinis) / (1-0)) %>%
  # filter(Slope >= 1 | Slope <= -1) %>%
  ggplot(., aes(x = species, y = Difference, fill = species)) + # Species
         geom_boxplot(outlier.colour = NA) +
         geom_jitter(aes(shape = species), alpha = 0.5, width = 0.2) +
         scale_fill_manual(values = sp_colors) +
         scale_shape_manual(values = c(4, 1)) +
         # scale_shape_manual(values = c(8, 3, 6, 1)) +
         theme_minimal(base_size = 12) +
         theme(axis.text.x = element_text(angle = 0, hjust = 0.5, size = 12),
               axis.text.y = element_text(size = 12)) +
         geom_hline(yintercept = 0, linetype = "dashed") +
         ylab("Difference") +
         xlab("Species")

```

```{r}

ac_ss_indep_map <- ac_indep_cov_plot + ac_indep_cov_box_plot +
  plot_layout(widths = c(1.5, 1)) +
  plot_layout(guides = "collect")

ac_ss_indep_map

# ggsave("./06-publication/main_figures/Figure_XX6/ss_independent_map_ac.pdf", plot = ac_ss_indep_map, device = "pdf",
#        scale = 1, width = 6, height = 4, units = c("in"), dpi = 300)

```


```{r, eval = F}
# sinensis
vcf_list_sinensis_ac <- list.files(path = "./04-analysis/sinensis_sanguinis_mapping/ancient_calculus/sinensis", pattern = ".vcf$", full.names = T)

vcf_ac_sinensis_sum_snp <- map_dfr(vcf_list_sinensis_ac, function(fn) {
  fread(fn, col.names = c("Contig", "POS", "ID","REF","ALT", "QUAL", "FILTER","INFO", "FORMAT","BAM"), sep = "\t", skip = 147, header = F) %>%
  mutate(Library_ID = basename(fn)) %>%
  as_tibble() %>%
  filter(FILTER == "PASS")
}) %>%
# remove the end of the file name from the Library_ID
  mutate(Library_ID = str_replace_all(Library_ID, ".sinensis.vcf","")) 

# fwrite(vcf_ac_sinensis_sum_snp, "05-results/vcf_ac_sinensis_snp.tsv", quote = F, sep = "\t")

# sanguinis
vcf_list_sanguinis_ac <- list.files(path = "./04-analysis/sinensis_sanguinis_mapping/ancient_calculus/sanguinis", pattern = ".vcf$", full.names = T)

vcf_ac_sanguinis_sum_snp <- map_dfr(vcf_list_sanguinis_ac, function(fn) {
  fread(fn, col.names = c("Contig", "POS", "ID","REF","ALT", "QUAL", "FILTER","INFO", "FORMAT","BAM"), sep = "\t", skip = 31, header = F) %>%
  mutate(Library_ID = basename(fn)) %>%
  as_tibble() %>%
  filter(FILTER == "PASS")
}) %>%
# remove the end of the file name from the Library_ID
  mutate(Library_ID = str_replace_all(Library_ID, ".sanguinis.vcf",""))

# fswrite(vcf_ac_sanguinis_sum_snp, "05-results/vcf_ac_sanguinis_snp.tsv", quote = F, sep = "\t")

```

```{r}
vcf_ac_sinensis_snp <- fread("../strep_clades/05-results/vcf_ac_sinensis_snp.tsv.gz") %>%
  as_tibble()
vcf_ac_sanguinis_snp <- fread("../strep_clades/05-results/vcf_ac_sanguinis_snp.tsv.gz") %>%
  as_tibble()

vcf_ac_sinensis_sum_snp <- vcf_ac_sinensis_snp %>%
  full_join(., genome_metadata, by = "Contig") %>%
  drop_na(Library_ID) %>%
  separate(INFO, into = "Depth", extra = "drop", sep = ";") %>%
  mutate(Depth = str_replace_all(Depth, "DP=",""),
         Depth = as.numeric(Depth)) %>% 
  filter(Depth >= 2) %>%
  group_by(Library_ID, Assembly_accession) %>%
  count() %>%
  ungroup() %>%
  left_join(., genome_metadata %>%
              select(Assembly_accession, Species, GTDB_species), by = "Assembly_accession") %>%
  distinct()


vcf_ac_sanguinis_sum_snp <- vcf_ac_sanguinis_snp %>%
  full_join(., genome_metadata, by = "Contig") %>%
  drop_na(Library_ID) %>%
  separate(INFO, into = "Depth", extra = "drop", sep = ";") %>%
  mutate(Depth = str_replace_all(Depth, "DP=",""),
         Depth = as.numeric(Depth)) %>% 
  filter(Depth >= 2) %>%
  group_by(Library_ID, Assembly_accession) %>%
  count() %>%
  ungroup() %>%
  left_join(., genome_metadata %>%
              select(Assembly_accession, Species, GTDB_species), by = "Assembly_accession") %>%
  distinct()

# combine together into 1 file
vcf_ac_independent_snp <- vcf_ac_sinensis_sum_snp %>%
  bind_rows(., vcf_ac_sanguinis_sum_snp) %>%
  rename(no_snps_2X_cov = n)

```

```{r}
vcf_ac_independent_snp %>%
  left_join(., ac_most_abund_strep %>%
               separate(Species, into = "species", sep = "_", extra = "drop"), by = "Library_ID") %>%
  filter(str_detect(species, "sinensis|sanguinis")) %>%
  left_join(., genome_lengths) %>%
  mutate(proportion_snps = no_snps_2X_cov / GenomeLen,
         Assembly_accession = fct_relevel(Assembly_accession, sp_order)) %>%
  arrange(Assembly_accession) %>%
  ggplot(., aes(x = Assembly_accession, y = proportion_snps, fill = GTDB_species)) +
         geom_boxplot(outlier.colour = NA) +
         geom_jitter(aes(shape = species), alpha = 0.5, width = 0.2) +
         scale_fill_manual(values = sp_colors) +
         scale_shape_manual(values = c(4, 1)) +
         # scale_shape_manual(values = c(8, 3, 6, 1)) +
         theme_minimal(base_size = 10) +
         theme(axis.text.x = element_text(angle = 90, hjust = 1)) +
         xlab("Assembly accession") +
         ylab("% SNPs/Genome") +
         ggtitle("Independent genome mapping")

```

```{r}
vcf_ac_independent_snp %>%
  left_join(., ac_most_abund_strep %>%
               separate(Species, into = "species", sep = "_", extra = "drop"), by = "Library_ID") %>%
  filter(str_detect(species, "sinensis|sanguinis")) %>%
  left_join(., genome_lengths) %>%
  mutate(proportion_snps = no_snps_2X_cov / GenomeLen,
         Assembly_accession = fct_relevel(Assembly_accession, sp_order)) %>%
  rename(Assembly_species = GTDB_species) %>%
  arrange(Library_ID) %>%
  ggplot(.) +
         geom_point(aes(x = Assembly_species, y = proportion_snps, fill = species), shape = 21, color = "black") + # , fill = "#766df8"
         geom_line(aes(x = Assembly_species, y = proportion_snps, group = Library_ID), linetype="dotted") +
         theme_minimal(base_size = 12) +
         scale_fill_manual(values = sp_colors) +
         theme(axis.text.x = element_text(angle = 0, hjust = 0.5, size = 12),
               axis.text.y = element_text(size = 12)) +
         xlab("Genome") +
         ylab("Proportion of SNPs covered at 2X") +
         facet_wrap(~species)

```

```{r}
# Make a list of the Iberian project samples
iberian_list <- lib_metadata %>%
  filter(Study == "Iberian") %>%
  select(Study, Sample_alias) %>%
  mutate(Sample_alias = str_replace_all(Sample_alias, ".SG1.1","")) %>%
  filter(!str_detect(Sample_alias, "ELR|IVE")) %>%
  pull(Sample_alias)

```

```{r multiallelic_snps, eval = F}
# This dataset has the Iberian calculus dataset and we need to remove it. 

# sinensis
tsv_list_sinensis_ac <- list.files(path = "./04-analysis/sinensis_sanguinis_mapping/ancient_calculus/sinensis", pattern = ".vcf.tsv$", full.names = T)

tsv_ac_sinensis_sum_snp <- map_dfr(tsv_list_sinensis_ac, function(fn) {
  fread(fn, col.names = c("Contig", "POS", "REF", "DP", "ALT"), sep = "\t") %>%
  mutate(Library_ID = basename(fn)) %>%
  as_tibble()
}) %>%
# remove the end of the file name from the Library_ID
  mutate(Library_ID = str_replace_all(Library_ID, ".sinensis.vcf.tsv","")) %>%
  filter(!str_detect(Library_ID, iberian_list %>% str_c(collapse = "|")))

# fwrite(tsv_ac_sinensis_sum_snp, "05-results/tsv_ac_sinensis_snp.tsv", sep = "\t", quote = F)

# sanguinis
tsv_list_sanguinis_ac <- list.files(path = "./04-analysis/sinensis_sanguinis_mapping/ancient_calculus/sanguinis", pattern = ".vcf.tsv$", full.names = T)

tsv_ac_sanguinis_sum_snp <- map_dfr(tsv_list_sanguinis_ac, function(fn) {
  fread(fn, col.names = c("Contig", "POS", "REF", "DP", "ALT"), sep = "\t") %>%
  mutate(Library_ID = basename(fn)) %>%
  as_tibble()
}) %>%
# remove the end of the file name from the Library_ID
  mutate(Library_ID = str_replace_all(Library_ID, ".sanguinis.vcf.tsv","")) %>%
  filter(!str_detect(Library_ID, iberian_list %>% str_c(collapse = "|")))

# fwrite(tsv_ac_sanguinis_sum_snp, "05-results/tsv_ac_sanguinis_snp.tsv", sep = "\t", quote = F)

```

```{r}
tsv_ac_sinensis_snp <- fread("../strep_clades/05-results/tsv_ac_sinensis_snp.tsv.gz") %>%
  as_tibble()
tsv_ac_sanguinis_snp <- fread("../strep_clades/05-results/tsv_ac_sanguinis_snp.tsv.gz") %>%
  as_tibble()

#. sinensis
tsv_ac_sinensis_sum_snp <- tsv_ac_sinensis_snp %>%
  group_by(Library_ID, Contig, POS) %>%
  count() %>%
  ungroup() %>%
  left_join(., genome_metadata %>%
              select(Assembly_accession, Contig, Species, GTDB_species), by = "Contig") %>%
  distinct()

# sanguinis
tsv_ac_sanguinis_sum_snp <- tsv_ac_sanguinis_snp %>%
  group_by(Library_ID, Contig, POS) %>%
  count() %>%
  ungroup() %>%
  left_join(., genome_metadata %>%
              select(Assembly_accession, Contig, Species, GTDB_species), by = "Contig") %>%
  distinct()

```

```{r multiallelic_snps_test_ct}

# sinensis
tsv_ac_sinensis_ct_snp <- tsv_ac_sinensis_snp %>%
  mutate(CT = (REF == "C" & ALT == "T")) %>%
  filter(CT == F) %>%
  group_by(Library_ID, Contig, POS) %>%
  count(name = "no_alleles") %>%
  ungroup() %>%
  left_join(., genome_metadata %>%
              select(Assembly_accession, Contig, Species, GTDB_species), by = "Contig") %>%
  distinct()

# sanguinis

tsv_ac_sanguinis_ct_snp <- tsv_ac_sanguinis_snp %>%
  mutate(CT = (REF == "C" & ALT == "T")) %>%
  filter(CT == F) %>%
  group_by(Library_ID, Contig, POS) %>%
  count(name = "no_alleles") %>%
  ungroup() %>%
  left_join(., genome_metadata %>%
              select(Assembly_accession, Contig, Species, GTDB_species), by = "Contig") %>%
  distinct()

```

```{r}
# total multi-allelic snps
tsv_ac_sinensis_ct_snp %>%
  mutate(Library_ID = str_replace_all(Library_ID, ".SG1", "")) %>%
  bind_rows(., tsv_ac_sanguinis_ct_snp %>%
              mutate(Library_ID = str_replace_all(Library_ID, ".SG1", ""))) %>%
  filter(no_alleles > 2) %>%
  group_by(Library_ID, Species) %>%
  count(name = "multiallelic_snps") %>%
  ungroup() %>%
  mutate(Species = str_replace_all(Species, "Streptococcus", "S.")) %>%
  left_join(., ac_most_abund_strep %>%
               separate(Species, into = "species", sep = "_", extra = "drop"), by = c("Library_ID")) %>%
  # some Pacific samples have no species
  drop_na(species) %>%
  ggplot(., aes(x = Species, y = multiallelic_snps, fill = Species)) +
         geom_boxplot(outlier.colour = NA) +
         geom_jitter(aes(shape = species), alpha = 0.5, width = 0.2) +
         scale_shape_manual(values = c(8, 3, 6, 1, 12)) +
         scale_fill_manual(values = c("#253494", "#C70E7B")) +
         theme_minimal(base_size = 10) +
         xlab("Assembly accession") +
         theme(axis.text.x = element_text(angle = 0, hjust = 0.5)) +
         ylab("# multi-allelic snps")

# % of multi-allelic snps out of all snps covered at least 2X
# maybe need to normalize this by genome length
tsv_ac_sinensis_ct_snp %>%
  mutate(Library_ID = str_replace_all(Library_ID, ".SG1", "")) %>%
  bind_rows(., tsv_ac_sanguinis_ct_snp %>%
              mutate(Library_ID = str_replace_all(Library_ID, ".SG1", ""))) %>%
  filter(no_alleles > 2) %>%
  group_by(Library_ID, Species) %>%
  count(name = "multiallelic_snps") %>%
  ungroup() %>%
  left_join(., vcf_ac_independent_snp %>%
              mutate(Library_ID = str_replace_all(Library_ID, ".SG1", "")), by = c("Library_ID","Species")) %>%
  mutate(wrong = multiallelic_snps > no_snps_2X_cov) %>%
  filter(wrong == FALSE,
         no_snps_2X_cov > 1) %>%
  select(-wrong) %>%
  mutate(pct_multiallelic_snps = multiallelic_snps / no_snps_2X_cov * 100) %>%
  left_join(., ac_most_abund_strep %>%
               separate(Species, into = "species", sep = "_", extra = "drop"), by = "Library_ID") %>%
  filter(str_detect(species, "sinensis|sanguinis")) %>%
  mutate(Species = str_replace_all(Species, "Streptococcus", "S.")) %>%
  ggplot(., aes(x = Species, y = pct_multiallelic_snps, fill = Species)) +
         geom_boxplot(outlier.colour = NA) +
         geom_jitter(aes(shape = species), alpha = 0.5, width = 0.2) +
         scale_fill_manual(values = c("#253494", "#C70E7B")) +
         scale_shape_manual(values = c(4, 1)) +
         # scale_shape_manual(values = c(8, 3, 6, 1)) +
         theme_minimal(base_size = 10) +
         theme(axis.text.x = element_text(angle = 90, hjust = 1)) +
         xlab("Assembly accession") +
         ylab("% SNPs covered min 2X that are multiallelic")


```

### Ancient Calculus Coverage with MAGs

```{r ac_mag_load_cov, eval = F}

sss_list_mag_ac <- list.files(path = "./04-analysis/sinensis_sanguinis_mapping/deep_evo_mags/ancient_calculus/", pattern = ".cov", full.names = T)

sss_ac_mag_cov <- map_dfr(sss_list_mag_ac, function(fn) {
  fread(fn, col.names = c("Contig", "Depth", "Breadth","Genome_length","Proportion_covered"), sep = "\t") %>%
  mutate(Library_ID = basename(fn)) %>%
  as_tibble()
}) %>%
  arrange(Library_ID) %>%
  filter(Contig != "genome") %>%
  # now add the genome accession to the table for filtering
  # remove the end of the file name from the Library_ID
  mutate(Library_ID = str_replace_all(Library_ID, "-mag_ss.mapped_rmdup.cov","")) 

# fwrite(sss_ac_mag_cov, "05-results/sss_ac_mag_cov.tsv", quote = F, sep = "\t")

```


```{r ac_mag_load_cov}
sss_ac_mag_cov <- fread("../strep_clades/05-results/sss_ac_mag_cov.tsv.gz") %>%
  as_tibble()

# read in the mag metadata table
mag_metadata <- fread("./00-documentation/deep_evo_mag_metadata.tsv")

sss_ac_mag_cov <- sss_ac_mag_cov %>%
  inner_join(., genome_metadata %>%
              bind_rows(., mag_metadata)) %>%
  # full_join(., mag_metadata, by = c("Assembly_accession","Contig", "Species","GTDB_species")) %>%
  select(Assembly_accession, everything()) 


sss_ac_mag_cov %>%
  filter(Library_ID == "2084", 
         str_detect(Assembly_accession, "JAE012"))


```


```{r ac_mag_genome_coverages}

Genomes <- genome_metadata %>%
  filter(str_detect(Assembly_accession, "GCF_013343115.1|GCF_000767835.1")) %>%
  bind_rows(., mag_metadata) %>%
  select(Assembly_accession) %>%
  unique() %>%
  pull()

tot_genome_length <- fread("./00-documentation/mag_ss_strep.fa.fai", sep = "\t", col.names = c("Contig","genome_length","other1","other2","other3")) %>%
  as_tibble() %>%
  left_join(., mag_metadata) %>%
  drop_na(Assembly_accession) %>%
  group_by(Assembly_accession) %>%
  summarize(GenomeLen = sum(genome_length)) %>%
  bind_rows(., genome_lengths)

ac_mag_genomes_coverage <- sss_ac_mag_cov %>%
  filter(Assembly_accession != "all") %>%
  filter(Depth > 0) %>%
  group_by(Library_ID, Assembly_accession) %>%
  summarize(bases_cov_min1X = sum(Breadth)) %>%
  ungroup() %>%
  full_join(., sss_ac_mag_cov %>%
              filter(Assembly_accession != "all") %>%
               select(Assembly_accession, Library_ID, Depth, Breadth) %>%
               filter(Depth >= 2) %>%
               group_by(Library_ID, Assembly_accession) %>%
               summarize(bases_cov_min2X = sum(Breadth)) %>%
               ungroup()) %>%
  full_join(., sss_ac_mag_cov %>%
              filter(Assembly_accession != "all") %>%
               select(Assembly_accession, Library_ID, Depth, Breadth) %>%
               filter(Depth >= 5) %>%
               group_by(Library_ID, Assembly_accession) %>%
               summarize(bases_cov_min5X = sum(Breadth)) %>%
               ungroup()) %>%
  full_join(., tot_genome_length, by = "Assembly_accession") %>%
  group_by(Library_ID, Assembly_accession) %>%
  summarize(breadth_1X = bases_cov_min1X / GenomeLen * 100,
            breadth_2X = bases_cov_min2X / GenomeLen * 100,
            breadth_5X = bases_cov_min5X / GenomeLen * 100) %>%
  ungroup() %>%
  pivot_longer(!Library_ID:Assembly_accession, names_to = "Depth", values_to = "Breadth")

# Genomes <- c("GCF_013343115.1","GCF_000767835.1")

ac_mag_genomes_coverage %>%
  select(Assembly_accession) %>%
  unique()

```


```{r ac_mag_coverages}
# species with max breadth of coverage at 1X depth
ac_mag_genomes_coverage %>%
  filter(Depth == "breadth_1X") %>%
  group_by(Library_ID) %>%
  slice_max(Breadth) %>%
  left_join(., genome_metadata %>%
              bind_rows(., mag_metadata) %>%
              select(Assembly_accession, GTDB_species)) %>%
  distinct()
  
# species with max breadth of coverage at 2X depth
ac_mag_genomes_coverage %>%
  filter(Depth == "breadth_2X") %>%
  group_by(Library_ID) %>%
  slice_max(Breadth) %>%
  left_join(., genome_metadata %>%
              bind_rows(., mag_metadata) %>%
              select(Assembly_accession, GTDB_species)) %>%
  distinct() %>%
  # remove CMC032.B b/c it has no coverage at 2X depth (poor sample)
  filter(Library_ID != "CMC032.B0101.SG1")

# species with max breadth of coverage at 5X depth
ac_mag_genomes_coverage %>%
  filter(Depth == "breadth_5X") %>%
  group_by(Library_ID) %>%
  slice_max(Breadth) %>%
  left_join(., genome_metadata %>%
              bind_rows(., mag_metadata) %>%
              select(Assembly_accession, GTDB_species)) %>%
  distinct() %>%
  # remove CMC032.B b/c it has no coverage at 5X depth (poor sample)
  filter(Library_ID != "CMC032.B0101.SG1")
 
```

```{r}
ac_mag_genomes_coverage_plot <- ac_mag_genomes_coverage %>%
  filter(Depth != "breadth_2X") %>%
  left_join(., genome_metadata %>%
              select(Assembly_accession, GTDB_species)) %>%
  distinct() %>%
  mutate(Library_ID = str_replace_all(Library_ID, ".SG1", "")) %>%
  full_join(., ac_most_abund_strep %>%
               separate(Species, into = "species", sep = "_", extra = "drop"), by = "Library_ID") %>%
  filter(!is.na(species)) %>%
  mutate(Assembly_accession = fct_relevel(Assembly_accession, sp_order)) %>%
  filter(!is.na(Breadth)) %>%
  ggplot(., aes(x = Assembly_accession, y = Breadth, fill = Assembly_accession)) + # Species
         geom_boxplot(outlier.colour = NA) +
         geom_jitter(aes(shape = species), alpha = 0.5, width = 0.2) +
         scale_shape_manual(values = c(8, 6, 4, 1, 12)) +
 # figure out how to reorder Accessions so they're in the right order (now sinensis and sanguinis are switched)
         scale_fill_manual(values = c("#253494", "#C70E7B", "#EC921D", "#f9ca7f", "#fbe1b6", "#fef7ec")) +
         ylim(0,100) +
         theme_minimal(base_size = 10) +
         theme(axis.text.x = element_text(angle = 90, hjust = 1, vjust = 0.5)) +
         ylab("Breadth of genome covered (%)") +
         xlab("Assembly accession") +
         facet_wrap(~Depth)

ac_mag_genomes_coverage_plot

ggsave("./06-publication/supplemental_figures/Sup_fig_S16/ac_mag_genomes_coverage_plot.pdf", plot = ac_mag_genomes_coverage_plot, device = "pdf",
       scale = 1, width = 8, height = 5, units = c("in"), dpi = 300)


```






