---
title: "Strep species mapping"
author: "Irina Velsko"
date: "`r format(Sys.time(), '%b %d, %Y')`"
output:
    github_document:
        fig_width: 8
        fig_height: 6
        toc: true
---

```{r load_libraries, echo = F, message = F}
library(knitr)
library(data.table)
library(janitor)
library(dtplyr)
library(tidyverse)
library(patchwork)
opts_chunk$set(echo=F, warning=F, message=F, dpi=150)
```

Set the working directory of the notebook to the parent directory of the 
notebook itself.  

```{r setup}
knitr::opts_knit$set(root.dir = normalizePath("../../.."))
```

```{r}
sp_colors <- c("sanguinis" = "#ffffd9", "sanguinis_A" = "#edf8b1", "sanguinis_C" = "#c7e9b4", "sanguinis_D" = "#7fcdbb", "sanguinis_E" = "#41b6c4", "sanguinis_F" = "#1d91c0", "sanguinis_G" = "#225ea8", "sanguinis_H" = "#253494", "sanguinis_I" = "#081d58", "sinensis" = "#C70E7B")
```


```{r load_metadata}

genome_metadata <- fread("./00-documentation/sinensis_sanguinis_contigs.tsv") %>%
  as_tibble()

# also need to normalize the number of SNPs by number of bps in each genome
genome_lengths <- fread("./00-documentation/S_sinensis_sanguinis.fna.fai", sep = "\t", header = FALSE) %>%
  as_tibble() %>%
  rename(Contig = 1,
         Length = 2,
         TotalLen = 3,
         Offset = 4,
         Offset_1 = 5) %>%
  full_join(., genome_metadata %>%
              select(Contig, Assembly_accession), by = "Contig") %>%
  group_by(Assembly_accession) %>%
  summarize(GenomeLen = sum(Length)) %>%
  ungroup()

sp_order <- genome_lengths %>%
  left_join(., genome_metadata, by = "Assembly_accession") %>%
  select(Assembly_accession, GTDB_species) %>%
  distinct() %>%
  arrange(GTDB_species) %>%
  pull(Assembly_accession)


#  
lib_metadata <- fread("./00-documentation/strep_clades_metadata.tsv") %>%
  as_tibble() %>%
  filter(!str_detect(Study, "ObregonTito2015|Oh2016|Rampelli2015|Sankaranarayanan2015|Slon2017")) %>%
    filter(Sample_group != "non_human_primate",
           !str_detect(Sample_group,"urban_gut|blank|Bone")
           ) %>%
  mutate(Secondary_sample_accession = ifelse(Study == "Velsko2018", Sample_alias,Secondary_sample_accession),
         Secondary_sample_accession = ifelse(Study == "Velsko2023", Sample_alias,Secondary_sample_accession),
         Secondary_sample_accession = ifelse(Study == "Velsko2022", Sample_alias,Secondary_sample_accession),
         Secondary_sample_accession = ifelse(Study == "FellowsYates2021", Sample_alias,Secondary_sample_accession),
         Secondary_sample_accession = ifelse(Study == "Clemente2015", Run_accession,Secondary_sample_accession),
         Secondary_sample_accession = ifelse(Study == "Lassalle2017", Run_accession,Secondary_sample_accession),
         Secondary_sample_accession = ifelse(Study == "Brealey2020", Sample_alias,Secondary_sample_accession),
         Secondary_sample_accession = ifelse(Study == "Eerkens2018", Run_accession,Secondary_sample_accession),
         Secondary_sample_accession = ifelse(Study == "Eisenhofer2020", Run_accession,Secondary_sample_accession),
         Secondary_sample_accession = ifelse(Study == "Fagernaes2022", Sample_alias,Secondary_sample_accession),
         Secondary_sample_accession = ifelse(Study == "Granehaell2021", Sample_alias,Secondary_sample_accession),
         Secondary_sample_accession = ifelse(Study == "Jacobson2020", Sample_alias,Secondary_sample_accession),
         Secondary_sample_accession = ifelse(Study == "Mann2018", Sample_alias,Secondary_sample_accession),
         Secondary_sample_accession = ifelse(Study == "Mann2018_w", Run_accession,Secondary_sample_accession),
         Secondary_sample_accession = ifelse(Study == "Ottoni2019", Sample_alias,Secondary_sample_accession),
         Secondary_sample_accession = ifelse(Study == "Ottoni2021", Run_accession,Secondary_sample_accession),
         Secondary_sample_accession = ifelse(Study == "Ozga2019", Sample_alias,Secondary_sample_accession),
         Secondary_sample_accession = ifelse(Study == "Ottoni2021", Run_accession,Secondary_sample_accession),
         Secondary_sample_accession = ifelse(Study == "Iberian", Sample_alias,Secondary_sample_accession),
         Secondary_sample_accession = ifelse(Study == "Pacific", Sample_alias,Secondary_sample_accession),
         Secondary_sample_accession = ifelse(Study == "Weyrich2017", Run_accession,Secondary_sample_accession),
         Secondary_sample_accession = ifelse(Study == "Warinner2014", Sample_alias,Secondary_sample_accession),
         Secondary_sample_accession = ifelse(Study == "Neukamm2020", Sample_alias,Secondary_sample_accession)) %>%
  mutate(Secondary_sample_accession = str_replace_all(Secondary_sample_accession, ".SG1.1","")) %>%
  mutate(Full = ifelse(Study == "Clemente2015","Buccal mucosa - Non-industrial",
                       ifelse(Study == "Lassalle2017","Saliva - Non-industrial",
                              ifelse(Study == "Velsko2023", "Plaque - Non-industrial",
                                     ifelse(Study == "Velsko2018","Calculus - Industrial",
                                            ifelse(Study == "FellowsYates2021","Calculus - Industrial","Industrial")))))) %>%
  mutate(Full = ifelse(Study == "HMP2012" & str_detect(Sample_type, "plaque"),"Plaque - Industrial",
                       ifelse(Study == "HMP2012" & Sample_type == "saliva","Saliva - Industrial",
                              ifelse(Study  ==  "HMP2012" & Sample_type == "buccal_mucosa","Buccal mucosa - Industrial",
                                     ifelse(Study ==  "Brito2016","Saliva - Industrial",Full))))) %>%
  mutate(Full = fct_relevel(Full, "Plaque - Industrial","Plaque - Non-industrial","Calculus - Industrial","Buccal mucosa - Industrial","Buccal mucosa - Non-industrial","Saliva - Industrial","Saliva - Non-industrial"))


# also read in cmc metadata for plots later
cmc_metadata <- fread("https://raw.githubusercontent.com/ivelsko/Cameroon_plaque/master/00-documentation.backup/01-cameroon_hunter_gatherer_metadata.tsv") %>%
  as_tibble() %>%
  rename(Library_ID = 1) %>%
  filter(Env == "CMC") %>%
  select(Library_ID, Ethnic_Group, Market_integration, Sex, Tooth_site)

```

```{r}
# file with species names and clade designations
gtdb_clades <- fread("./00-documentation/dRep_gtdb_clades.tsv") %>%
  as_tibble()

gtdb_clades <- gtdb_clades %>%
  mutate(Clade = fct_relevel(Clade, "Sanguinis","Mitis","Salivarius","Anginosus","Bovis","Pyogenic","Mutans","Downei","Other","Unknown"))

```


## CMC

```{r load_data}
# read in poor samples determined by cuperdec
poor_samples <- fread("./05-results/cuperdec_anc_calc_poor_samples.tsv") %>%
  # there are no poorly preserved modern plaque or calculus samples
  pull(Sample)

# read in the species table
mod_oral_raw <- fread("./05-results/mod_calc.kraken2.gtdb.report_mpa.tsv") %>%
  full_join(., fread("./05-results/ind_plaque.kraken2.gtdb.report_mpa.tsv"), by = "#Classification") %>%
  full_join(., fread("./05-results/ind_buccal_mucosa.kraken2.gtdb.report_mpa.tsv"), by = "#Classification") %>%
  full_join(., fread("./05-results/ind_saliva.kraken2.gtdb.report_mpa.tsv"), by = "#Classification") %>%
  full_join(., fread("./05-results/nonind_plaque.kraken2.gtdb.report_mpa.tsv"), by = "#Classification") %>%
  full_join(., fread("./05-results/nonind_buccal_mucosa.kraken2.gtdb.report_mpa.tsv"), by = "#Classification") %>%
  full_join(., fread("./05-results/nonind_saliva.kraken2.gtdb.report_mpa.tsv"), by = "#Classification") %>%
  replace(is.na(.), 0)

# read in the file names to add as column headers
mod_names <- fread("./05-results/mod_calc_names.tsv", header = F) %>%
  bind_rows(., fread("./05-results/indpq_names.tsv", header = F)) %>%
  bind_rows(., fread("./05-results/indbm_names.tsv", header = F)) %>%
  bind_rows(., fread("./05-results/indsal_names.tsv", header = F)) %>%
  bind_rows(., fread("./05-results/nonindpq_names.tsv", header = F)) %>%
  bind_rows(., fread("./05-results/nonindbm_names.tsv", header = F)) %>%
  bind_rows(., fread("./05-results/nonindsal_names.tsv", header = F)) %>%
  pull()

# add the sample names to the table
# move the lineage to the rownames so only samples are left as columns
mod_oral_raw <- mod_oral_raw %>%
  as_tibble() %>%
  column_to_rownames("#Classification")

# add the sample names
colnames(mod_oral_raw) <- mod_names

# move the lineage back to a column
mod_oral_raw <- mod_oral_raw %>%
  rownames_to_column("Lineage")

# filter to have only Streptococcus species
mod_oral_raw_sp <- mod_oral_raw %>% 
  filter(str_detect(Lineage, "s__s__")) %>%
  separate(Lineage, into = c("domain","phylum","class","order","family","genus","Species"), sep = "\\|", extra = "merge") %>%
  select(-c("domain","phylum","class","order","family","genus")) %>%
  mutate(Species = str_replace_all(Species, "s__s__","")) %>%
  filter(str_detect(Species, "Streptococcus"))

```

```{r}
# which samples have which strep species as the most abundant?
cmc_most_abund_strep <- mod_oral_raw_sp %>%
  pivot_longer(!Species, names_to = "Library_ID", values_to = "Counts") %>%
  left_join(., gtdb_clades, by = "Species") %>%
  filter(str_detect(Clade, "Sanguinis"),
         str_detect(Library_ID, "CMC")) %>%
  group_by(Library_ID) %>%
  slice_max(Counts) %>%
  ungroup() %>%
  select(-Counts, -Clade) %>%
  mutate(Species = str_replace_all(Species, "Streptococcus ", ""))

```

```{r cmc_genome_coverages}

Genomes <- genome_metadata %>%
  select(Assembly_accession) %>%
  unique() %>%
  pull()

```

### Independent mappings
```{r}

sinensis_cmc_cov <-  fread("05-results/sinensis_cmc_indep_cov.tsv.gz") %>%
  as_tibble()
sanguinis_cmc_cov <-  fread("05-results/sanguinis_cmc_indep_cov.tsv.gz") %>%
  as_tibble()

# now add the genome accession to the table for filtering
sinensis_cmc_cov <- sinensis_cmc_cov %>%
  left_join(., genome_metadata, by = "Contig") %>%
  select(Assembly_accession, everything()) %>%
  filter(Contig != "genome")


# now add the genome accession to the table for filtering
sanguinis_cmc_cov <- sanguinis_cmc_cov %>%
  left_join(., genome_metadata, by = "Contig") %>%
  select(Assembly_accession, everything()) %>%
  filter(Contig != "genome")

```


```{r cmc_independent_coverages}

cmc_independent_coverages <- sinensis_cmc_cov %>%
  filter(Assembly_accession != "all") %>%
  filter(Depth > 0) %>%
  group_by(Library_ID, Assembly_accession) %>%
  summarize(bases_cov_min1X = sum(Breadth)) %>%
  ungroup() %>%
  full_join(., sinensis_cmc_cov %>%
              filter(Assembly_accession != "all") %>%
               select(Assembly_accession, Library_ID, Depth, Breadth) %>%
               filter(Depth >= 2) %>%
               group_by(Library_ID, Assembly_accession) %>%
               summarize(bases_cov_min2X = sum(Breadth)) %>%
               ungroup()) %>%
  full_join(., sinensis_cmc_cov %>%
              filter(Assembly_accession != "all") %>%
               select(Assembly_accession, Library_ID, Depth, Breadth) %>%
               filter(Depth >= 5) %>%
               group_by(Library_ID, Assembly_accession) %>%
               summarize(bases_cov_min5X = sum(Breadth)) %>%
               ungroup()) %>%
  left_join(., genome_lengths, by = "Assembly_accession") %>%
  group_by(Library_ID, Assembly_accession) %>%
  summarize(breadth_1X = bases_cov_min1X / GenomeLen * 100,
            breadth_2X = bases_cov_min2X / GenomeLen * 100,
            breadth_5X = bases_cov_min5X / GenomeLen * 100) %>%
  ungroup() %>%
  bind_rows(., sanguinis_cmc_cov %>%
            filter(Assembly_accession != "all") %>%
            filter(Depth > 0) %>%
            group_by(Library_ID, Assembly_accession) %>%
            summarize(bases_cov_min1X = sum(Breadth)) %>%
            ungroup() %>%
            full_join(., sanguinis_cmc_cov %>%
                         filter(Assembly_accession != "all") %>%
                         select(Assembly_accession, Library_ID, Depth, Breadth) %>%
                         filter(Depth >= 2) %>%
                         group_by(Library_ID, Assembly_accession) %>%
                         summarize(bases_cov_min2X = sum(Breadth)) %>%
                         ungroup()) %>%
            full_join(., sanguinis_cmc_cov %>%
                         filter(Assembly_accession != "all") %>%
                         select(Assembly_accession, Library_ID, Depth, Breadth) %>%
                         filter(Depth >= 5) %>%
                         group_by(Library_ID, Assembly_accession) %>%
                         summarize(bases_cov_min5X = sum(Breadth)) %>%
                         ungroup()) %>%
  left_join(., genome_lengths, by = "Assembly_accession") %>%
  group_by(Library_ID, Assembly_accession) %>%
  summarize(breadth_1X = bases_cov_min1X / GenomeLen * 100,
            breadth_2X = bases_cov_min2X / GenomeLen * 100,
            breadth_5X = bases_cov_min5X / GenomeLen * 100) %>%
  ungroup())

```

```{r cov_independent}

cmc_indep_cov_plot <- cmc_independent_coverages %>%
  mutate(Library_ID = str_replace_all(Library_ID, ".SG1","")) %>%
  select(-breadth_2X, -breadth_5X) %>%
  left_join(., genome_metadata %>%
              select(Assembly_accession, GTDB_species) %>%
              rename(Assembly_species = GTDB_species)) %>%
  distinct() %>%
  full_join(., cmc_most_abund_strep %>%
               separate(Species, into = "species", sep = "_", extra = "drop"), by = "Library_ID") %>%
  filter(str_detect(species, "sinensis|sanguinis")) %>%
  drop_na(breadth_1X) %>%
  # get rid of the species cluster info so have only sanguinis/sinensis for plotting
  separate(Assembly_species, into = "Assembly_species", sep = "_", extra = "drop") %>%
  filter(Library_ID != "CMC032.B0101") %>%
  ggplot(.) +
         geom_point(aes(x = Assembly_species, y = breadth_1X, fill = species), shape = 21, color = "black") + # , fill = "#766df8"
         geom_line(aes(x = Assembly_species, y = breadth_1X, group = Library_ID), linetype="dotted") +
         theme_minimal(base_size = 12) +
         scale_fill_manual(values = sp_colors) +
         theme(axis.text.x = element_text(angle = 0, hjust = 0.5, size = 12),
               axis.text.y = element_text(size = 12)) +
         xlab("Genome") +
         ylab("% of genome covered at 1X") +
         facet_wrap(~species)
cmc_indep_cov_plot

```

```{r}
cmc_indep_cov_box_plot<-  cmc_independent_coverages %>%
  mutate(Library_ID = str_replace_all(Library_ID, ".SG1","")) %>%
  select(-breadth_2X, -breadth_5X) %>%
  left_join(., genome_metadata %>%
              select(Assembly_accession, GTDB_species) %>%
              rename(Assembly_species = GTDB_species)) %>%
  distinct() %>%
  full_join(., cmc_most_abund_strep %>%
               separate(Species, into = "species", sep = "_", extra = "drop"), by = "Library_ID") %>%
  filter(str_detect(species, "sinensis|sanguinis")) %>%
  drop_na(breadth_1X) %>%
  # get rid of the species cluster info so have only sanguinis/sinensis for plotting
  separate(Assembly_species, into = "Assembly_species", sep = "_", extra = "drop") %>%
  select(-Assembly_accession) %>%
  filter(Library_ID != "CMC032.B0101") %>%
  pivot_wider(names_from = "Assembly_species", values_from = "breadth_1X") %>%
  mutate(Difference = sanguinis - sinensis,
         Slope = (sinensis - sanguinis) / (1-0)) %>%
  # filter(Slope >= 1 | Slope <= -1) %>%
  ggplot(., aes(x = species, y = Slope, fill = species)) + # Species
         geom_boxplot(outlier.colour = NA) +
         geom_jitter(aes(shape = species), alpha = 0.5, width = 0.2, size = 3) +
         scale_fill_manual(values = sp_colors) +
         scale_shape_manual(values = c(4, 1)) +
         # scale_shape_manual(values = c(8, 3, 6, 1)) +
         theme_minimal(base_size = 12) +
         theme(axis.text.x = element_text(angle = 0, hjust = 0.5, size = 12),
               axis.text.y = element_text(size = 12)) +
         geom_hline(yintercept = 0, linetype = "dashed") +
         ylab("Slope") +
         xlab("Species") 
cmc_indep_cov_box_plot

```

```{r}

cmc_ss_indep_map <- cmc_indep_cov_plot + cmc_indep_cov_box_plot +
  plot_layout(widths = c(1.5, 1)) +
  plot_layout(guides = "collect")

cmc_ss_indep_map

# ggsave("./06-publication/main_figures/Figure_5/ss_independent_map_cmc.pdf", plot = cmc_ss_indep_map, device = "pdf",
#        scale = 1, width = 6, height = 4, units = c("in"), dpi = 300)

```


### Kraken2 GTDB r202 table
```{r}
# log-transform the data
hm_strep_sp_modern <- mod_oral_raw_sp  %>%
  # remove poorly-preserved samples
  select(-matches(poor_samples %>% str_c(collapse = "|"))) %>%
  # remove CMC032.B0101 b/c it's poorly preserved
  select(-matches("CMC032.B0101|SRR1646041")) %>%
  arrange(Species) %>%
  # convert counts to proportions to normalize across all studies
  adorn_percentages(denominator = "col") %>%
  left_join(., gtdb_clades, by = "Species") %>%
  mutate(Species = str_replace_all(Species, "Streptococcus","S.")) %>%
  filter(str_detect(Clade, "Anginosus|Sanguinis"),
         !str_detect(Species, "_|sp")) %>%
  select(-Clade) %>%
  pivot_longer(!Species, names_to = "Library_ID", values_to = "Percent") %>%
  mutate(Percent = Percent * 100) %>%
  # add 1 for log transform
  mutate(Percent = Percent + 1) %>%
  mutate(logPercent = log10(Percent)) %>%
  left_join(., lib_metadata %>%
              rename(Library_ID = Secondary_sample_accession) %>%
              select(Library_ID, Full), by = "Library_ID") %>%
  filter(str_detect(Full, "Plaque|Calculus")) %>%
  mutate(
    # Library_ID = fct_relevel(Library_ID, sample_order_rs),
         Library_ID = str_replace_all(Library_ID, "VLC","aVLC"),
         Library_ID = str_replace_all(Library_ID, "JAE", "aJAE"),
         Species = fct_relevel(Species, "S. cristatus", "S. sanguinis","S. gordonii","S. sinensis","S. constellatus","S. anginosus","S. intermedius")) %>%
  arrange(Species) %>%
  ggplot(., aes(Species, Library_ID, fill = logPercent)) +
    geom_tile() +
    scale_fill_distiller(palette = "Blues", direction=-1) +
    # scale_fill_gradient2(midpoint = 0, low = "#4A6FE3", mid = "#f7f7f7", high = "#D33F6A") +
    theme_minimal(base_size = 3) +
    theme(
          axis.text.y = element_blank(),
          # axis.text.x = element_text(angle = 90, hjust = 1, vjust = 0.5),
          axis.text.x = element_text(size = 3)) +
    theme(plot.margin = margin(0, 0, 0, 0, "pt"),
          axis.title.y = element_blank(),
          legend.position = "left") +
     scale_y_discrete(limits=rev) +
     ylab("Percent") +
     xlab("Sample")
# +
#      facet_grid(vars(Full), scales = "free")

hm_strep_sp_modern

# save w/o sample names
# ggsave("./06-publication/main_figures/Figure_5/hm_strep_species_modern.pdf", plot = hm_strep_sp_modern, device = "pdf",
#        scale = 1, width = 2, height = 6, units = c("in"), dpi = 300)

# saave with sample names
# ggsave("./06-publication/main_figures/Figure_5/hm_strep_species_modern_samples.pdf", plot = hm_strep_sp_modern, device = "pdf",
#        scale = 1, width = 3, height = 8, units = c("in"), dpi = 300)

```

## Ancient calculus

```{r load_data}
# read in the species table. It has the column names already
anc_calc_raw <- fread("./05-results/anc_calc.kraken2.gtdb.report_mpa.tsv.gz") %>%
  as_tibble()

anc_names <- fread("./05-results/anc_calc_names.tsv", header = F)  %>%
  as_tibble() %>%
  rename(Library_ID = 1) %>%
  mutate(Library_ID = str_replace_all(Library_ID, ".SG1.1", ""),
         Library_ID = str_replace_all(Library_ID, ".SG1", ""))

anc_calc_strep_sp_pre <- anc_calc_raw %>% 
  filter(str_detect(Lineage, "s__s__")) %>%
  separate(Lineage, into = c("domain","phylum","class","order","family","genus","Species"), sep = "\\|", extra = "merge") %>%
  select(-c("domain","phylum","class","order","family","genus")) %>%
  mutate(Species = str_replace_all(Species, "s__s__","")) %>%
  filter(str_detect(Species, "Streptococcus"))

# filter to have only Streptococcus
# these samples have very, very few reads that were assigned to the Sanguinis group (0-36), so we'll remove them
ac_strep_multiples <- anc_calc_strep_sp_pre %>%
  pivot_longer(!Species, names_to = "Library_ID", values_to = "Counts") %>%
  left_join(., gtdb_clades, by = "Species") %>%
  filter(str_detect(Clade, "Sanguinis")) %>%
  group_by(Library_ID) %>%
  slice_max(Counts) %>%
  ungroup() %>%
  select(-Counts, -Clade) %>%
  mutate(Species = str_replace_all(Species, "Streptococcus ", "")) %>%
  group_by(Library_ID) %>%
  count() %>% 
  ungroup() %>%
  filter(n > 1) %>%
  pull(Library_ID)

anc_calc_strep_sp <- anc_calc_raw %>% 
  filter(str_detect(Lineage, "s__s__")) %>%
  separate(Lineage, into = c("domain","phylum","class","order","family","genus","Species"), sep = "\\|", extra = "merge") %>%
  select(-c("domain","phylum","class","order","family","genus")) %>%
  mutate(Species = str_replace_all(Species, "s__s__","")) %>%
  filter(str_detect(Species, "Streptococcus")) %>%
  # remove Radcliffe soil and dentin
  select(-matches(ac_strep_multiples %>% str_c(collapse = "|")))

```

```{r}
# which samples have which strep species as the most abundant?
anc_calc_strep_sp %>%
  pivot_longer(!Species, names_to = "Library_ID", values_to = "Counts") %>%
  left_join(., gtdb_clades, by = "Species") %>%
  filter(str_detect(Clade, "Sanguinis")) %>%
  group_by(Library_ID) %>%
  slice_max(Counts) %>%
  ungroup() %>%
  arrange(Species) %>%
  group_by(Species) %>%
  count()

ac_most_abund_strep <- anc_calc_strep_sp %>%
  pivot_longer(!Species, names_to = "Library_ID", values_to = "Counts") %>%
  left_join(., gtdb_clades, by = "Species") %>%
  filter(str_detect(Clade, "Sanguinis")) %>%
  arrange(Library_ID, Species) %>%
  group_by(Library_ID) %>%
  slice_max(Counts) %>%
  ungroup() %>%
  select(-Counts, -Clade) %>%
  mutate(Species = str_replace_all(Species, "Streptococcus ", "")) 

```

```{r ac_load_independent_cov}
sinensis_ac_cov <- fread("./05-results/sinensis_ac_indep_cov.tsv.gz") %>%
  as_tibble()
sanguinis_ac_cov <- fread("./05-results/sanguinis_ac_indep_cov.tsv.gz") %>%
  as_tibble()

# now add the genome accession to the table for filtering
sinensis_ac_cov <- sinensis_ac_cov %>%
  left_join(., genome_metadata, by = "Contig") %>%
  select(Assembly_accession, everything()) %>%
  filter(Contig != "genome")


# now add the genome accession to the table for filtering
sanguinis_ac_cov <- sanguinis_ac_cov %>%
  left_join(., genome_metadata, by = "Contig") %>%
  select(Assembly_accession, everything()) %>%
  filter(Contig != "genome")

```


```{r ac_independent_coverages}

ac_independent_coverages <-  sinensis_ac_cov %>%
  filter(Assembly_accession != "all") %>%
  filter(Depth > 0) %>%
  group_by(Library_ID, Assembly_accession) %>%
  summarize(bases_cov_min1X = sum(Breadth)) %>%
  ungroup() %>%
  full_join(., sinensis_ac_cov %>%
              filter(Assembly_accession != "all") %>%
               select(Assembly_accession, Library_ID, Depth, Breadth) %>%
               filter(Depth >= 2) %>%
               group_by(Library_ID, Assembly_accession) %>%
               summarize(bases_cov_min2X = sum(Breadth)) %>%
               ungroup()) %>%
  full_join(., sinensis_ac_cov %>%
              filter(Assembly_accession != "all") %>%
               select(Assembly_accession, Library_ID, Depth, Breadth) %>%
               filter(Depth >= 5) %>%
               group_by(Library_ID, Assembly_accession) %>%
               summarize(bases_cov_min5X = sum(Breadth)) %>%
               ungroup()) %>%
  full_join(., genome_lengths, by = "Assembly_accession") %>%
  group_by(Library_ID, Assembly_accession) %>%
  summarize(breadth_1X = bases_cov_min1X / GenomeLen * 100,
            breadth_2X = bases_cov_min2X / GenomeLen * 100,
            breadth_5X = bases_cov_min5X / GenomeLen * 100) %>%
  ungroup() %>%
  bind_rows(., sanguinis_ac_cov %>%
  filter(Assembly_accession != "all") %>%
  filter(Depth > 0) %>%
  group_by(Library_ID, Assembly_accession) %>%
  summarize(bases_cov_min1X = sum(Breadth)) %>%
  ungroup() %>%
  full_join(., sanguinis_ac_cov %>%
              filter(Assembly_accession != "all") %>%
               select(Assembly_accession, Library_ID, Depth, Breadth) %>%
               filter(Depth >= 2) %>%
               group_by(Library_ID, Assembly_accession) %>%
               summarize(bases_cov_min2X = sum(Breadth)) %>%
               ungroup()) %>%
  full_join(., sanguinis_ac_cov %>%
              filter(Assembly_accession != "all") %>%
               select(Assembly_accession, Library_ID, Depth, Breadth) %>%
               filter(Depth >= 5) %>%
               group_by(Library_ID, Assembly_accession) %>%
               summarize(bases_cov_min5X = sum(Breadth)) %>%
               ungroup()) %>%
  full_join(., genome_lengths, by = "Assembly_accession") %>%
  group_by(Library_ID, Assembly_accession) %>%
  summarize(breadth_1X = bases_cov_min1X / GenomeLen * 100,
            breadth_2X = bases_cov_min2X / GenomeLen * 100,
            breadth_5X = bases_cov_min5X / GenomeLen * 100) %>%
  ungroup()) %>%
  pivot_longer(!Library_ID:Assembly_accession, names_to = "Depth", values_to = "Breadth")


```

```{r cov_independent}

ac_indep_cov_plot <-  ac_independent_coverages %>%
  mutate(Library_ID = str_replace_all(Library_ID, ".SG1","")) %>%
  filter(Depth == "breadth_1X") %>%
  left_join(., genome_metadata %>%
              select(Assembly_accession, GTDB_species) %>%
              rename(Assembly_species = GTDB_species)) %>%
  distinct() %>%
  full_join(., ac_most_abund_strep %>%
               separate(Species, into = "species", sep = "_", extra = "drop"), by = "Library_ID") %>%
  filter(str_detect(species, "sinensis|sanguinis")) %>%
  drop_na(Breadth) %>%
  # get rid of the species cluster info so have only sanguinis/sinensis for plotting
  separate(Assembly_species, into = "Assembly_species", sep = "_", extra = "drop") %>%
  ggplot(.) +
         geom_point(aes(x = Assembly_species, y = Breadth, fill = species), shape = 21, color = "black") + # , fill = "#766df8"
         geom_line(aes(x = Assembly_species, y = Breadth, group = Library_ID), linetype="dotted") +
         theme_minimal(base_size = 12) +
         scale_fill_manual(values = sp_colors) +
         theme(axis.text.x = element_text(angle = 0, hjust = 0.5, size = 12),
               axis.text.y = element_text(size = 12)) +
         xlab("Genome") +
         ylab("% of genome covered at 1X") +
         facet_wrap(~species)
ac_indep_cov_plot

ac_indep_cov_box_plot <- ac_independent_coverages %>%
  mutate(Library_ID = str_replace_all(Library_ID, ".SG1","")) %>%
  filter(Depth == "breadth_1X") %>%
  left_join(., genome_metadata %>%
              select(Assembly_accession, GTDB_species) %>%
              rename(Assembly_species = GTDB_species)) %>%
  distinct() %>%
  full_join(., ac_most_abund_strep %>%
               separate(Species, into = "species", sep = "_", extra = "drop"), by = "Library_ID") %>%
  filter(str_detect(species, "sinensis|sanguinis")) %>%
  drop_na(Breadth) %>%
  # get rid of the species cluster info so have only sanguinis/sinensis for plotting
  separate(Assembly_species, into = "Assembly_species", sep = "_", extra = "drop") %>%
  select(-Assembly_accession) %>%
  pivot_wider(names_from = "Assembly_species", values_from = "Breadth") %>%
  mutate(Difference = sinensis - sanguinis,
         Slope = (sinensis - sanguinis) / (1-0)) %>%
  # filter(Slope >= 1 | Slope <= -1) %>%
  ggplot(., aes(x = species, y = Slope, fill = species)) + # Species
         geom_boxplot(outlier.colour = NA) +
         geom_jitter(aes(shape = species), alpha = 0.5, width = 0.2, size = 3) +
         scale_fill_manual(values = sp_colors) +
         scale_shape_manual(values = c(4, 1)) +
         # scale_shape_manual(values = c(8, 3, 6, 1)) +
         theme_minimal(base_size = 12) +
         theme(axis.text.x = element_text(angle = 0, hjust = 0.5, size = 12),
               axis.text.y = element_text(size = 12)) +
         geom_hline(yintercept = 0, linetype = "dashed") +
         ylab("Slope") +
         xlab("Species") 
ac_indep_cov_box_plot

ac_independent_coverages %>%
  mutate(Library_ID = str_replace_all(Library_ID, ".SG1","")) %>%
  filter(Depth == "breadth_1X") %>%
  left_join(., genome_metadata %>%
              select(Assembly_accession, GTDB_species) %>%
              rename(Assembly_species = GTDB_species)) %>%
  distinct() %>%
  full_join(., ac_most_abund_strep %>%
               separate(Species, into = "species", sep = "_", extra = "drop"), by = "Library_ID") %>%
  filter(str_detect(species, "sinensis|sanguinis")) %>%
  drop_na(Breadth) %>%
  # get rid of the species cluster info so have only sanguinis/sinensis for plotting
  separate(Assembly_species, into = "Assembly_species", sep = "_", extra = "drop") %>%
  select(-Assembly_accession) %>%
  pivot_wider(names_from = "Assembly_species", values_from = "Breadth") %>%
  mutate(Difference = sinensis - sanguinis,
         Slope = (sinensis - sanguinis) / (1-0)) %>%
  # filter(Slope >= 1 | Slope <= -1) %>%
  ggplot(., aes(x = species, y = Difference, fill = species)) + # Species
         geom_boxplot(outlier.colour = NA) +
         geom_jitter(aes(shape = species), alpha = 0.5, width = 0.2, size = 3) +
         scale_fill_manual(values = sp_colors) +
         scale_shape_manual(values = c(4, 1)) +
         # scale_shape_manual(values = c(8, 3, 6, 1)) +
         theme_minimal(base_size = 12) +
         theme(axis.text.x = element_text(angle = 0, hjust = 0.5, size = 12),
               axis.text.y = element_text(size = 12)) +
         geom_hline(yintercept = 0, linetype = "dashed") +
         ylab("Difference") +
         xlab("Species")

```

```{r}

ac_ss_indep_map <- ac_indep_cov_plot + ac_indep_cov_box_plot +
  plot_layout(widths = c(1.5, 1)) +
  plot_layout(guides = "collect")

ac_ss_indep_map

# ggsave("./06-publication/main_figures/Figure_5/ss_independent_map_ac.pdf", plot = ac_ss_indep_map, device = "pdf",
#        scale = 1, width = 6, height = 4, units = c("in"), dpi = 300)

```


### Kraken2 GTDB r202 table
```{r load_data}
# read in the species table. It has the column names already
anc_calc_raw <- fread("./05-results/anc_calc.kraken2.gtdb.report_mpa.tsv.gz") 

anc_names <- fread("./05-results/anc_calc_names.tsv", header = F)  %>%
  rename(Library_ID = 1) %>%
  mutate(Library_ID = str_replace_all(Library_ID, ".SG1.1", ""),
         Library_ID = str_replace_all(Library_ID, ".SG1", ""))

# filter to have only species and remove all species with <1000 reads assigned
# even filtering out taxa with <10000 counts doesn't help
anc_calc_strep_sp <- anc_calc_strep_sp_pre %>% 
  # remove Radcliffe soil and dentin
  dplyr::select(-matches("CSS|CSD")) 

sample_order_rs <- anc_calc_strep_sp  %>%
  # remove poorly-preserved samples
  select(-matches(poor_samples %>% str_c(collapse = "|"))) %>%
  # join with the Strep grup table for plotting
  inner_join(., gtdb_clades) %>%
  select(-Species) %>%
  select(Clade, everything()) %>%
  # long-format
  pivot_longer(!Clade, names_to = "Library_ID", values_to = "Counts") %>%
  # sum together all species per group
  group_by(Library_ID, Clade) %>%
  summarize(Reads = sum(Counts)) %>%
  ungroup() %>%
  # transpose
  pivot_wider(names_from = "Clade", values_from = "Reads") %>%
  # convert table to proportions
  adorn_percentages(denominator = "row") %>%
  # full_join(., poor_samples %>% as_tibble(.) %>% rename(Library_ID = 1) %>% mutate(Preservation = "Poor")) %>%
  # mutate(Preservation = ifelse(is.na(Preservation), "Good",Preservation)) %>%
  # arrange by Sanguinis and for the ones w/o Sanguinis, by Anginosus
  # arrange(Preservation, Library_ID)
  arrange(desc(Sanguinis), Anginosus) %>% # Preservation, 
  pull(Library_ID)

```

```{r}
hm_strep_sp_ac <- anc_calc_strep_sp %>%
  # remove poorly-preserved samples
  select(-matches(poor_samples %>% str_c(collapse = "|"))) %>%
  # convert counts to proportions to normalize across all studies
  adorn_percentages(denominator = "col") %>%
  as_tibble() %>%
  full_join(., gtdb_clades, by = "Species") %>%
  select(Clade, Species, everything()) %>%
  filter(str_detect(Clade, "Sanguinis|Anginosus"),
         !str_detect(Species, "_|sp")) %>%  
  select(-Clade) %>%
  pivot_longer(!Species, names_to = "Library_ID", values_to = "Percent") %>%
  # convert proportion from adorn_percentages to a percent
  mutate(Percent = Percent * 100) %>%
  mutate(Species = str_replace_all(Species, "Streptococcus","S.")) %>%
  # remove poorly-preserved samples
  filter(!str_detect(Library_ID, poor_samples %>% str_c(collapse = "|"))) %>%
  # add 1 for log transform
  mutate(Percent = Percent + 1) %>%
  mutate(logPercent = log10(Percent)) %>%
  mutate(Library_ID = fct_relevel(Library_ID, sample_order_rs),
         Species = fct_relevel(Species, "S. cristatus", "S. sanguinis","S. gordonii","S. sinensis","S. constellatus","S. anginosus","S. intermedius")) %>%
  arrange(Species, Library_ID) %>%
  ggplot(., aes(Species, Library_ID, fill = logPercent)) +
    geom_tile() +
    scale_fill_distiller(palette = "Blues", direction=-1) +
    # scale_fill_gradient2(midpoint = 0, low = "#4A6FE3", mid = "#f7f7f7", high = "#D33F6A") +
    theme_minimal(base_size = 8) +
    theme(axis.text.y = element_blank()) +
    # theme(axis.text.x = element_text(angle = 90, hjust = 1, vjust = 0.5)) +
    theme(plot.margin = margin(0, 0, 0, 0, "pt"),
          axis.title.y = element_blank(),
          legend.position = "left") +
     scale_y_discrete(limits=rev) +
     ylab("Percent") +
     xlab("Sample")

hm_strep_sp_ac

# ggsave("./06-publication/main_figures/Figure_5/hm_strep_species_ac.pdf", plot = hm_strep_sp_ac, device = "pdf",
#        scale = 1, width = 2, height = 7, units = c("in"), dpi = 300)


```

```{r}

# ancient calc
dom_clade_lib <- fread("./05-results/anc_calc_dominant_clade_per_library.tsv")

anc_calc_strep_sp %>%
  # remove poorly-preserved samples
  select(-matches(poor_samples %>% str_c(collapse = "|"))) %>%
  # convert counts to proportions to normalize across all studies
  adorn_percentages(denominator = "col") %>%
  as_tibble() %>%
  full_join(., gtdb_clades, by = "Species") %>%
  select(Clade, Species, everything()) %>%
  filter(str_detect(Clade, "Sanguinis|Anginosus"),
         !str_detect(Species, "_|sp")) %>%  
  select(-Clade) %>%
  pivot_longer(!Species, names_to = "Library_ID", values_to = "Percent") %>%
  # convert proportion from adorn_percentages to a percent
  mutate(Percent = Percent * 100) %>%
  mutate(Species = str_replace_all(Species, "Streptococcus","S.")) %>%
  # remove poorly-preserved samples
  filter(!str_detect(Library_ID, poor_samples %>% str_c(collapse = "|"))) %>%
  group_by(Library_ID) %>%
  slice_max(Percent) %>%
  ungroup() %>%
  full_join(., dom_clade_lib %>%
              select(-Percent), by = "Library_ID") %>%
  # arrange(Clade, Species)
  group_by(Clade) %>%
  count(Species) %>%
  ungroup()


dom_clade_lib_mod <- fread("./05-results/mod_dominant_clade_per_library.tsv")

# modern CMC plaque
mod_oral_raw_sp  %>%
  # remove poorly-preserved samples
  select(-matches(poor_samples %>% str_c(collapse = "|"))) %>%
  # remove CMC032.B0101 b/c it's poorly preserved
  select(-matches("CMC032.B0101|SRR1646041")) %>%
  arrange(Species) %>%
  # convert counts to proportions to normalize across all studies
  adorn_percentages(denominator = "col") %>%
  left_join(., gtdb_clades, by = "Species") %>%
  mutate(Species = str_replace_all(Species, "Streptococcus","S.")) %>%
  filter(str_detect(Clade, "Anginosus|Sanguinis"),
         !str_detect(Species, "_|sp")) %>%
  select(-Clade) %>%
  pivot_longer(!Species, names_to = "Library_ID", values_to = "Percent") %>%
  mutate(Percent = Percent * 100) %>%
  mutate(Species = str_replace_all(Species, "Streptococcus","S.")) %>%
  # only the CMC samples
  filter(str_detect(Library_ID, "CMC")) %>%
  # select(Library_ID) %>% unique()
  group_by(Library_ID) %>%
  slice_max(Percent) %>%
  ungroup() %>%
  left_join(., dom_clade_lib_mod %>%
              select(-Percent), by = "Library_ID") %>%
  # arrange(Clade, Species)
  group_by(Clade) %>%
  count(Species) %>%
  ungroup()

# modern HMP plaque
mod_oral_raw_sp  %>%
  # remove poorly-preserved samples
  select(-matches(poor_samples %>% str_c(collapse = "|"))) %>%
  # remove CMC032.B0101 b/c it's poorly preserved
  select(-matches("CMC032.B0101|SRR1646041")) %>%
  arrange(Species) %>%
  # convert counts to proportions to normalize across all studies
  adorn_percentages(denominator = "col") %>%
  left_join(., gtdb_clades, by = "Species") %>%
  mutate(Species = str_replace_all(Species, "Streptococcus","S.")) %>%
  filter(str_detect(Clade, "Anginosus|Sanguinis"),
         !str_detect(Species, "_|sp")) %>%
  select(-Clade) %>%
  pivot_longer(!Species, names_to = "Library_ID", values_to = "Percent") %>%
  mutate(Percent = Percent * 100) %>%
  mutate(Species = str_replace_all(Species, "Streptococcus","S.")) %>%
  # only the HMP plaque samples
  left_join(., lib_metadata %>%
              select(Secondary_sample_accession, Full) %>%
              rename(Library_ID = 1), by = "Library_ID") %>%
  filter(str_detect(Full, "Plaque - Industrial")) %>%
  unique() %>%
  # select(Library_ID) %>% unique()
  group_by(Library_ID) %>%
  slice_max(Percent) %>%
  ungroup() %>%
  # left_join(., dom_clade_lib_mod %>%
  #             select(-Percent), by = "Library_ID") %>%
  # arrange(Clade, Species)
  group_by(Species) %>%
  count() %>%
  # ungroup() %>%
  arrange(Species)

```


